.. image:: /Images/NC_State.gif
   :target: http://www.ncsu.edu


Semester Overview, 2020
=======================


.. image:: /Images/syllabus_logo.jpg

Image credit: `Lilian Matallana <https://www.linkedin.com/in/lilian-matallana-21704474/>`_




			**Class meetings are in Room 6117 Jordan Hall from 8:30 to 10:20 am on Mondays, Wednesdays, and Fridays.**

				The `Biostar Handbook <https://www.biostarhandbook.com/>`_ is a resource for much of the reading.

				The `Piazza online forum page <https://piazza.com/ncsu/spring2019/bit815/>`_ is available for questions and discussion.

				Teaching assistant: `Will Kohlway <whkohlwa@ncsu.edu>`_



Course Schedule 
***************
::

	  Date			Topic [Biostar Handbook Chapter]

	6–10 Jan	Introduction to Linux and the command-line interface [1, 2, 4]

	13–17 Jan	Sequencing instruments, Experimental design, 
	 	        	 Data preprocessing and quality control [8, 9, 10]

	20 Jan		Martin Luther King holiday, no class	 

	22-24 Jan	Error correction and alignment	 

	27-31 Jan	Assembly - transcriptomes and genomes [22]

	3-7 Feb		Re-sequencing, alignment, structural variation [17, 18, 23]

	11-15 Feb	Discovery and genotyping of genetic variation	 

	17-21 Feb	R and R Studio - lectures and exercises through Software 	 
		         	 Carpentry website, sections 1 - 8

	24-28 Feb	R and R Studio, continued - sections 9 - 12. 
		            	 Advanced: Data Carpentry for Genomics

	2-6 Mar		Transcriptome analysis: differential gene expression, 
		       	 annotation [19, 20]

	9-13 Mar	Spring break - no classes	 

	16-20 Mar	Genome analysis: ChIP-seq, DHS-seq, 3-D conformation	 

	23-27 Mar	Linux command-line tools: awk, sed, and bash [15]

	30 Mar-3 Apr	CLC Genomics Workbench - data QC and pre-processing	 

	6-10 Apr	CLC Genomics Workbench - RNA-seq analysis	 

	13-17 Apr	CLC Genomics Workbench - genome assembly tutorial	 

	20-22 Apr	CLC Genomics Workbench - variant analysis	 





General background information and course resources
***************************************************


+	General advice on `troubleshooting <troubleshooting.html>`_
+	`Course syllabus <https://drive.google.com/open?id=1HgYJGXv_mxbKycTMrtK_dg37jjuEAvVG>`_
+	Class bash history files from instructor VCL instances
+	Lior Pachter's list of sequencing-based assays: `\*Seq <https://liorpachter.wordpress.com/seq/>`_
+	`The R statistical programming environment <r-materials.html>`_
+	`Course resources <resources.html>`_
+	`Overview slides <https://drive.google.com/open?id=10RYNwJXx7gwYCA_o_1u8AtRw465ROjZn>`_
+	Cloud computing with Amazon Web Services: `setting up an AWS account <https://drive.google.com/open?id=1OXA_TAYu2l_--GEAW85eKJCLUtWyqhbN>`_ and `starting an instance <https://drive.google.com/open?id=1U7D7BRfS1LLbWGzJwkBejc8vfyRSPLIc>`_
+	OMICtools software database `publication <http://database.oxfordjournals.org/content/2014/bau069.long>`_ and `website <http://omictools.com/>`_
+	Qiagen webpage of `tutorials <https://www.qiagenbioinformatics.com/support/tutorials/>`_ for CLC workbench programs
+	`Colib'read project webpage <https://colibread.inria.fr/project/>`_ on reference-assembly-free programs for SNP and indel detection and moress 


A flow-chart overview of DNA sequencing experiments
***************************************************

	.. image:: /Images/Flowchart.jpg 







Last modified 8 December 2019.
Edits by `Ross Whetten <https://github.com/rwhetten>`_, `Will Kohlway <https://github.com/wkohlway>`_, & `Maria Adonay <https://github.com/amalgamaria>`_.
