.. image:: /Images/NC_State.gif
   :target: http://www.ncsu.edu	


CS 590-001, Sequence Analysis for Plant Genetics
======================================

Instructors: 
 * Dr. Joseph Gage
 * Dr. Ryan Sartor
****************************

Course Description
------------------
The materials for this course were forked and modified from Dr. Ross Whetten's Deep Sequence Analysis (BIT 815) course at NCSU. You can access the original materials at https://deepseqanalysis.readthedocs.io/en/latest/

Class sessions consist of 75 minutes twice per week. On Tuesdays, we will generally begin with a short lecture and Q&A on the week's topic before beginning hands-on computing exercises using model datasets. Thursday's class will allow more time for unstructured Q&A and continuation of the hands-on exercise. 

The Sequence Analysis for Plant Genetics course is designed to introduce biologists to the Linux command-line computing environment and open-source software for analysis of next-generation sequencing data. We will emphasize the importance of cloud computing, due to the increasing demands for RAM and storage space required for analysis and storage of high-throughput sequencing data, and the cost-effectiveness and flexibility provided by cloud computing solutions. Applications of sequencing discussed include genome sequencing (both de-novo and resequencing), transcriptome analysis, discovery of sequence and structural variations, ChIP-seq methods for mapping DNA-protein interactions, and reduced representation genotyping (GBS and RAD-seq methods). For each application of sequencing technology, discussion topics include experimental design strategies, methods for library construction, sources of experimental and biological variation, and analytical approaches available in open-source software packages. Computing exercises utilize the software discussed, and provide participants with the opportunity to carry out analysis of sample datasets using a virtual machine image through the NC State University Virtual Computing Lab. This Linux system is customized to provide the bioinformatics software described during the course, and is available for class participants to use at any time. **The objective of the course is not to make course participants experts in every aspect of sequence analysis, but instead to empower participants to learn the specific skills they need by teaching basic skills in command-line Linux computing.** The course is directed at graduate students, but has also attracted participation from faculty, post-doctoral researchers, and research technicians interested in expanding their skills in the area of sequence data analysis.


.. toctree::
   :maxdepth: 2

   course-schedule  
   readings
   hardware
   intro-linux
   Sequencing Instruments <https://www.biostarhandbook.com/sequencing-instruments.html>
   experimental-design
   data-preprocessing-qc
   error-correct
   transcriptome-assembly 
   genome-assembly 
   re-seq-align-struct 
   genotyping
   r-lesson
   transcriptome-analysis
   chromatin-structure
   awk-sed-bash
   clc
   HPC




Indices and tables
==================

* :ref:`genindex`
* :ref:`modindex`
* :ref:`search`


Last modified 16 December 2022.
Edits by `Joe Gage <https://github.com/joegage>`_.
