.. image:: /Images/NC_State.gif
   :target: http://www.ncsu.edu 

.. image:: /Images/FERlogo.png
   :target: http://www.cnr.ncsu.edu/fer/

Readings and Resources
======================

Global overview books and papers
********************************

+ The Biostar Handbook - Bioinformatics Data Analysis Guide. Istvan Albert and others. Available `online <https://biostar.myshopify.com/>`_
+ Next generation quantitative genetics in plants. Jiménez-Gómez, Frontiers in Plant Science 2:77, 2011 `Full Text <http://www.frontiersin.org/Plant_Physiology/10.3389/fpls.2011.00077/full>`_ [*Equally relevant to animal and microbial systems*]
+ Sense from sequence reads: methods for alignment and assembly. Flicek & Birney, Nat Methods 6(11 Suppl):S6-S12, 2009. `Full Text <http://www.nature.com/nmeth/journal/v6/n11s/full/nmeth.1376.html>`_

Library construction and experimental design
********************************************

+ Statistical design and analysis of RNA sequencing data. Auer & Doerge, Genetics 185(2):405-16, 2010. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2881125>`_
+ Biases in Illumina transcriptome sequencing caused by random hexamer priming. Hansen et al., Nucleic Acids Res. 38(12): e131, 2010. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2896536>`_
+ Analyzing and minimizing PCR amplification bias in Illumina sequencing libraries. Aird et al, Genome Biology 12:R18, 2011 `Full Text <http://genomebiology.com/2011/12/2/R18>`_
+ Amplification-free Illumina sequencing-library preparation facilitates improved mapping and assembly of GC-biased genomes. Kozarewa et al, Nature Methods 6(4):291-295, 2009 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2664327/>`_
+ Cost-effective, high-throughput DNA sequencing libraries for multiplexed target capture. Rohland & Reich, Genome Research 22(5): 939–946, 2012. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3337438/>`_
+ Predicting the molecular complexity of sequencing libraries. Daley & Smith, Nature Methods 10(4):325-327, 2013 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3612374/>`_
+ RNA-seq differential expression studies: more sequence or more replication? Liu et al., Bioinformatics 30: 301 - 304, 2014. `Publisher Web Site <http://bioinformatics.oxfordjournals.org/content/30/3/301.long>`_
+ Power analysis and sample size estimation for RNA-seq differential expression. Ching et al., RNA 20: 1684 - 1696, 2014. `Publisher Web Site <http://rnajournal.cshlp.org/content/20/11/1684.short>`_
+ Guidance for RNA-seq co-expression network construction and analysis: safety in numbers. Ballouz et al., Bioinformatics 31: 2123 - 2130, 2014. `Publisher Web Site <http://bioinformatics.oxfordjournals.org/content/31/13/2123.short>`_
+ Points of significance: replication. Blainey et al., Nature Methods 11: 879–880, 2014. `Publisher Web Site <http://www.nature.com/nmeth/journal/v11/n9/full/nmeth.3091.html>`_
+ Points of Significance: Nested designs. Krzywinski et al., Nature Methods 11: 977–978, 2014 `Publisher Web Site <http://www.nature.com/nmeth/journal/v11/n10/full/nmeth.3137.html>`_
+ Points of significance: Sources of variation Altman & Krzywinski. Nature Methods 12: 5 – 6, 2015 `Publisher Web Site <http://www.nature.com/nmeth/journal/v12/n1/full/nmeth.3224.html>`_

Data formats, data management, and alignment software tools
***********************************************************
+ The Sequence Alignment/Map format and SAMtools. Li et al, Bioinformatics 25(16):2078-9, 2009 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2723002>`_
+ SAM format specification `file <http://samtools.sourceforge.net/SAM1.pdf>`_
+ Efficient storage of high throughput sequencing data using reference-based compression. Fritz et al, Genome Res 21(5):734-40, 2011. `Full Text <http://genome.cshlp.org/content/21/5/734.long>`_
+ Compression of DNA sequence reads in FASTQ format. Deorowicz & Grabowski, Bioinformatics 27(6):860-2, 2011. `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/21252073>`_
+ Fast and accurate short read alignment with Burrows-Wheeler transform. Li & Durbin, Bioinformatics 25(14):1754-60, 2009. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2705234>`_
+ Improving SNP discovery by base alignment quality. Li H, Bioinformatics 27(8):1157-8, 2011. `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/21320865>`_
+ BEDTools: a flexible suite of utilities for comparing genomic features. Quinlan and Hall, Bioinformatics 26:841-842, 2010. `Publisher Website <http://bioinformatics.oxfordjournals.org/content/26/6/841.full.pdf+html>`_

Data quality assessment, filtering, and correction
**************************************************
+ HTQC: a fast quality control toolkit for Illumina sequencing data. Yang et al, BMC Bioinformatics 14:33, 2013. `PubMed <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3571943/>`_
+ FastQC: a quality control tool for high-throughput sequence data. `Home Page <http://www.bioinformatics.bbsrc.ac.uk/projects/fastqc/>`_
+ FASTX-toolkit: FASTQ/A short-reads pre-processing tools `Home Page <http://hannonlab.cshl.edu/fastx_toolkit/>`_
+ QuorUM: an error corrector for Illumina reads.  Marçais et al. `2013 Arxiv preprint <http://arxiv.org/abs/1307.3515>`_ or `2015 PLoSOne paper <http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0130821>`_
+ Quake: quality-aware detection and correction of sequencing errors. Kelley et al, Genome Biol 11(11):R116, 2010. `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/21114842>`_
+ Reference-free validation of short read data. Schröder et al, PLoS One 5(9):e12681, 2010. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2943903>`_
+ Correction of sequencing errors in a mixed set of reads. Salmela, Bioinformatics 26(10):1284, 2010. `Full Text <http://bioinformatics.oxfordjournals.org/content/26/10/1284.long>`_ [*Includes error correction of SOLiD reads in colorspace.*]
+ Repeat-aware modeling and correction of short read errors. Yang et al, BMC Bioinformatics 12(Supp1):S52, 2011 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3044310>`_ [*Requires a reference sequence.*]
+ HiTEC: accurate error correction in high-throughput sequencing data. Ilie et al, Bioinformatics 27(3):295, 2011 `Full Text <http://bioinformatics.oxfordjournals.org/content/27/3/295.long>`_
+ Error correction of high-throughput sequencing datasets with non-uniform coverage. Medvedev et al., Bioinformatics 27(13):i137-41, 2011. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3117386>`_
+ Characterization of the Conus bullatus genome and its venom-duct transcriptome. Hu et al., BMC Genomics 12:60, 2011 `Full Text <http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-12-60>`_ [*Includes a novel strategy for estimating genome size from a partial transcriptome assembly and low-coverage (3x) genome sequence.*]

De novo assembly
****************

+ Velvet: algorithms for de novo short read assembly using de Bruijn graphs. Zerbino & Birney, Genome Res 18(5):821-9, 2008. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336801>`_
+ Assembly of large genomes using second-generation sequencing. Schatz et al, Genome Res 20(9):1165-73, 2010. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928494>`_
+ High-quality draft assemblies of mammalian genomes from massively parallel sequence data. Gnerre et al, PNAS 108(4): 1513-18, 2011 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3029755>`_
+ Genome assembly has a major impact on gene content: a comparison of annotation in two Bos taurus assemblies. Florea  et al., PLoS One 6(6):e21400, 2011. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3120881/>`_
+ Artemis: an integrated platform for visualization and analysis of high-throughput sequence-based experimental data. Carver et al, Bioinformatics 28(4):464 - 469, 2012 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3278759/>`_
+ Efficient de novo assembly of large genomes using compressed data structures. Simpson & Durbin, Genome Research 22:549-556, 2012 `Full Text <http://genome.cshlp.org/content/22/3/549.full>`_ [*Describes the String Graph Assembler (SGA), which assembled a human genome in less than 6 days using 54 Gb of RAM and a 123-processor compute cluster for calculation of an FM-index of the 1.2 billion reads*]
+ Readjoiner: a fast and memory efficient string graph-based sequence assembler. Gonnella & Kurtz, BMC Bioinformatics 13: 82, 2012 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3507659>`_
+ Assemblathon 1: A competitive assessment of de novo short read assembly methods. Earl et al, Genome Research 21:2224-2241, 2011 `Full Text <http://genome.cshlp.org/content/early/2011/09/16/gr.126599.111.full.pdf+html>`_

Chromatin analysis
******************

+ Bias Correction
  + Identifying and mitigating bias in next-generation sequencing methods for chromatin biology. Meyer and Liu, Nat Rev Genetics 15: 709 - 721, 2014 `Publisher Web Site <http://www.nature.com/nrg/journal/v15/n11/abs/nrg3788.html>`_

+ Chromatin Immunoprecipitation sequencing: ChIP-seq
  + ChIP-seq: advantages and challenges of a maturing technology. Park, Nat Rev Genet. 10:669-80, 2009 `PubMed <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3191340/>`_
  + ChIP-seq and Beyond: new and improved methodologies to detect and characterize protein-DNA interactions. Furey, Nat Rev Genet 13: 840–852, 2012 `Publisher Web Site <http://www.nature.com/nrg/journal/v13/n12/full/nrg3306.html>`_
  + MuMoD: a Bayesian approach to detect multiple modes of protein–DNA binding from genome-wide ChIP data. Narlikar, Nucleic Acids Res 41:21–32, 2013 `PubMed <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3592440/>`_

+ Chromatin conformation
  + A decade of 3C technologies: insights into nuclear organization. de Wit & de Laat, Genes & Devel  26: 11-24, 2012 `Publisher Website <http://genesdev.cshlp.org/content/26/1/11.full>`_
  + Exploring the three-dimensional organization of genomes: interpreting chromatin interaction data. Dekker et al, Nature Reviews Genetics 14: 390–403, 2013 `Publisher Website <http://www.nature.com/nrg/journal/v14/n6/full/nrg3454.html>`_

Transcriptome analysis
**********************

+ General considerations for RNA-seq library construction
  + Molecular indexing enables quantitative targeted RNA sequencing and reveals poor efficiencies in standard library preparations. Fu et al, PNAS 111:1891–1896, 2014 `Publisher Web Site <http://www.pnas.org/content/111/5/1891>`_

+ Assembly and comparison to genome
  + A glance at quality score: implication for de novo transcriptome reconstruction of Illumina reads. Mbandi et al., Frontiers in Genetics 2014. `Publisher Website <http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00017/full?>`_
  + Full-length transcriptome assembly from RNA-Seq data without a reference genome. Grabherr et al, Nature Biotechnology 29:644 - 652, 2011. `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/21572440>`_ [*Software called `Trinity <http://trinityrnaseq.sourceforge.net/>`_; available on Sourceforge.*]
  + Comprehensive analysis of RNA-Seq data reveals extensive RNA editing in a human transcriptome. Peng et al, Nature Biotechnology 30:253 - 260, 2012. `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/22327324>`_ [*Several comments on this paper question whether the reported differences are in fact evidence of editing or are simply sequencing errors - the authors stand by their conclusions, but the controversy demonstrates the importance of robust data analysis methods.*]
  + Optimization of de novo transcriptome assembly from next-generation sequencing data. Surget-Groba & Montoya-Burgos, Genome Res 20(10):1432-40, 2010. `Full Text <http://genome.cshlp.org/content/20/10/1432.long>`_
  + Rnnotator: an automated de novo transcriptome assembly pipeline from stranded RNA-Seq reads. Martin et al, BMC Genomics 11:663, 2010 `Full Text <http://www.biomedcentral.com/1471-2164/11/663>`_
  + De novo assembly and analysis of RNA-seq data. Robertson et al, Nature Methods 7:909-912, 2010 `Full Text <http://www.nature.com/nmeth/journal/v7/n11/full/nmeth.1517.html>`_ [*Describes Trans-ABySS, a pipeline to use the ABySS parallel assembler for de novo transcriptome analysis.*]

+ Differential expression analysis
  + Robust adjustment of sequence tag abundance. Baumann & Doerge, Bioinformatics 2013 `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/24108185>`_
  + R-SAP: a multi-threading computational pipeline for the characterization of high-throughput RNA-sequencing data. Mittal & McDonald, Nucleic Acids Res, 2012 `Full Text <http://nar.oxfordjournals.org/content/early/2012/01/28/nar.gks047.long>`_
  + Targeted RNA sequencing reveals the deep complexity of the human transcriptome. Mercer et al, Nature Biotechnology 30:99 - 104, 2012 `Publisher Website <http://www.nature.com/nbt/journal/v30/n1/full/nbt.2024.html>`_
  + Differential gene and transcript expression analysis of RNA-Seq experiments with TopHat and Cufflinks. Trapnell et al, Nature Protocols 7:562 - 578, 2012 `Publisher Website <http://www.nature.com/nprot/journal/v7/n3/full/nprot.2012.016.html>`_
  + Characterization and improvement of RNA-Seq precision in quantitative transcript expression profiling. Łabaj et al, Bioinformatics 27:i383 - i391, 2011 `Full Text <http://bioinformatics.oxfordjournals.org/content/27/13/i383.full.pdf+html>`_
  + Improving RNA-Seq expression estimates by correcting for fragment bias. Roberts et al, Genome Biol 12:R22, 2011 `PubMed Central <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3129672/>`_
  + Cloud-scale RNA-sequencing differential expression analysis with Myrna. Langmead et al, Genome Biol 11:R83, 2010 `Full Text <http://genomebiology.com/2010/11/8/R83>`_
  + From RNA-seq reads to differential expression results. Oshlack et al, Genome Biol 11(12):220, 2010 `Full Text <http://genomebiology.com/content/11/12/220>`_
  + DEGseq: an R package for identifying differentially expressed genes from RNA-seq data. Wang et al., Bioinformatics. 26(1):136-8. 2010 `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/19855105>`_
  + DEseq: Differential expression analysis for sequence count data. Anders and Huber, Genome Biology 11:R106, 2010 `Full Text <http://genomebiology.com/2010/11/10/R106>`_
  + Moderated estimation of fold change and dispersion for RNA-Seq data with DESeq2. Love et al, BioRxiv doi: 10.1101/002832, 2014 `Full Text <http://biorxiv.org/content/early/2014/02/19/002832>`_
  + edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Robinson et al., Bioinformatics 26(1):139-40 2010 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2796818>`_
  + Two-stage Poisson model for testing RNA-seq data. Auer and Doerge, SAGMB 10(1), article 26 `Full Text <http://www.bepress.com/sagmb/vol10/iss1/art26/>`_
  + Experimental design, preprocessing, normalization and differential expression analysis of small RNA sequencing experiments. McCormick et al., Silence2(1):2, 2011 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3055805>`_
  + RNA-Seq gene expression estimation with read mapping uncertainty. Li et al, Bioinformatics 26:493-500, 2010 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2820677>`_ [*Describes the RSEM software package.*]

Comparing genomes and assemblies; variant detection
***************************************************

  + Toward better understanding of artifacts in variant calling from high-coverage samples. Heng Li, Bioinformatics 30, 2843, 2014 `PubMedCentral <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4271055/>`_
  + Versatile and open software for comparing large genomes. Kurtz et al, Genome Biol (5(2):R12, 2004. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC395750>`_ [*Describes the MUMmer software for full-genome alignment & comparisons.*]
  + Searching for SNPs with cloud computing. Langmead et al, Genome Biol 10(11):R134, 2009 `Full Text <http://genomebiology.com/content/10/11/R134>`_
  + Calling SNPs without a reference sequence. Ratan et al, BMC Bioinformatics 11:130, 2010 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2851604>`_
  + Microindel detection in short-read sequence data. Krawitz et al, Bioinformatics 26(6):722-9, 2010. `Full Text <http://bioinformatics.oxfordjournals.org/content/26/6/722.long>`_
  + vipR: variant identification in pooled DNA using R. Altmann et al., Bioinformatics 27: i77-i84, 2011. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3117388>`_
  + Geoseq: a tool for dissecting deep-sequencing datasets. Gurtowski et al, BMC Bioinformatics 11:506, 2010. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2972303/>`_ [*Geoseq is a web service that allows searching deep sequencing datasets with a reference sequence of a gene of interest.*]
  + Detecting and annotating genetic variations using the HugeSeq pipeline. Lam et al, Nature Biotechnology 30:226 - 229, 2012 `Publisher Website <http://www.nature.com/nbt/journal/v30/n3/full/nbt.2134.html>`_, `Home Page <http://hugeseq.snyderlab.org/>`_
  + Genome-wide LORE1 retrotransposon mutagenesis and high-throughput insertion detection in Lotus japonicus. Urbański et al, Plant J 64:731-741, 2012. `Publisher Website <http://onlinelibrary.wiley.com/doi/10.1111/j.1365-313X.2011.04827.x/abstract>`_ [*This paper describes a 2-dimensional pooling strategy with barcoding to allow use of Illumina sequencing to screen for retrotransposon insertion mutations, and includes a software package called FSTpoolit for analysis of the resulting sequence reads.*]
  + Reproducibility of variant calls in replicate next-generation sequencing experiments. Qi et al., PLoS One 10: e0119230, 2015 `Full Text <http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0119230>`_

+ Genotyping by sequencing
  + Genome-wide genetic marker discovery and genotyping using next-generation sequencing. Davey et al., Nat Rev Genet 12(7):499-510, 2011 `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/21681211>`_ [*A review of methods available at the time.*]
  + A robust, simple genotyping-by-sequencing (GBS) approach for high diversity species. Elshire et al., PLoS One 6(5):e19379, 2011. `Full Text <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3087801>`_
  + Development of high-density genetic maps for barley and wheat using a novel two-enzyme genotyping-by-sequencing approach. Poland et al., PLoS One 7(2): e32253, 2012. `Full Text <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3289635/>`_
  + Double digest RADseq: an inexpensive method for de novo SNP discovery and genotyping in model and non-model species. Peterson et al, PLoS One 7(5):e37135, . 2012. `Full Text <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3365034/>`_
  + Imputation of unordered markers and the impact on genomic selection accuracy. Rutkowski et al, G3 3(3):427-39, 2013. `Full Text <http://www.g3journal.org/content/3/3/427.long>`_
  + Diversity Arrays Technology (DArT) and next-generation sequencing combined: genome-wide, high-throughput, highly informative genotyping for molecular breeding of Eucalyptus. Sansaloni et al., BMC Proceedings 5(Suppl 7):P54, 2011 `Full Text <http://www.biomedcentral.com/1753-6561/5/S7/P54>`_
  + High-throughput genotyping by whole-genome resequencing. Huang et al., Genome Res 19(6):1068-76, 2009. `Full Text <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2694477>`_
  + Multiplexed shotgun genotyping for rapid and efficient genetic mapping. Andolfatto et al. Genome Res 21(4):610-7, 2011. `Full Text <http://genome.cshlp.org/content/21/4/610.long>`_

+ Restriction-site Associated DNA (RAD) markers
  + Rapid SNP discovery and genetic mapping using sequenced RAD markers. Baird et al, PLoS One 3(10):e3376, 2008 `Full Text <http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0003376>`_
  + Linkage mapping and comparative genomics using next-generation RAD sequencing of a non-model organism. Baxter et al., PLoS One 6(4):e19315, 2011. `Full Text <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3082572>`_
  + Genome evolution and meiotic maps by massively parallel DNA sequencing: spotted gar, an outgroup for the teleost genome duplication. Amores et al, Genetics 188(4):799-808, 2011. `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/21828280>`_
  + Construction and application for QTL analysis of a Restriction-site Associated DNA (RAD) linkage map in barley. Chutimanitsakun et al, BMC Genomics 4; 12:4, 2011. `Full Text <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3023751>`_
  + RAD tag sequencing as a source of SNP markers in Cynara cardunculus L. Scaglione et al., BMC Genomics 13:3, 2012. `Full Text <http://www.biomedcentral.com/1471-2164/13/3>`
  + Paired-end RAD-seq for de novo assembly and marker design without available reference. Willing et al., Bioinformatics 27(16):2187-93, 2011. `Publisher Website <http://bioinformatics.oxfordjournals.org/content/27/16/2187.long>`_
  + Local de novo assembly of RAD paired-end contigs using short sequencing reads. Etter et al., PLOS ONE 6(4): e18561, 2011. `Full Text <http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0018561>`_
  + Stacks: building and genotyping loci de novo from short-read sequences. Catchen et al., G3: Genes, Genomes, Genetics, 1:171-182, 2011. `Home Page <http://creskolab.uoregon.edu/stacks/>`_
  + Rainbow: an integrated tool for efficient clustering and assembling RAD-seq reads. Chong et al, Bioinformatics 28(21):2732-7, 2012. `Publisher Website <http://bioinformatics.oxfordjournals.org/content/28/21/2732.long>`_
  + UK RAD Sequencing Wiki page, with bibliography and RADTools software download `Home Page <https://www.wiki.ed.ac.uk/display/RADSequencing/Home>`_

Population Genomics
*******************

+ PGDspider: an automated data conversion tool for connecting population genetics and genomics programs. Lischer & Excoffier, Bioinformatics 28: 298-299, 2012 `Publisher Website <http://bioinformatics.oxfordjournals.org/content/28/2/298.full>`_

Workspace environments
**********************

+ Papers
  + Galaxy: a comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences. Goecks et al, Genome Biol 11(8):R86, 2010 `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2945788>`_
  + Galaxy Cloudman: Delivering compute clusters. BMC Bioinformatics 11(Suppl. 12):S4, 2010 `Full Text <http://www.biomedcentral.com/content/pdf/1471-2105-11-S12-S4.pdf>`_
  + `The Genome Analysis Toolkit <http://www.broadinstitute.org/gsa/wiki/index.php/The_Genome_Analysis_Toolkit>`_: a MapReduce framework for analyzing next-generation DNA sequencing data. McKenna et al, Genome Res 20(9):1297-303, 2010. `PubMedCentral <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928508>`_
  + A framework for variation discovery and genotyping using next-generation DNA sequencing data. DePristo et al., Nat Genet 43(5):491-8, 2011. `PubMed <http://www.ncbi.nlm.nih.gov/pubmed/21478889>`_

+ Online resources
  + The `R statistical computing <http://cran.r-project.org/>`_ environment includes `Bioconductor <http://www.bioconductor.org/>`_, a specialized set of tools for analysis of microarray and high-throughput sequencing data. Introductory materials from on-line or short workshops are widely available online; examples are `Evomics2012 Bioconductor Tutorial <http://bioconductor.org/help/course-materials/2012/Evomics2012/Bioconductor-tutorial.pdf>`_, and `Intro to Bioconductor <http://bcb.dfci.harvard.edu/%7Eaedin/courses/Bioconductor/>`_. Materials from an advanced course on high-throughput genetic data analysis are at `Seattle 2012 materials <http://bioconductor.org/help/course-materials/2012/SeattleFeb2012/>`_. Thomas Girke of UC-Riverside has written a very complete set of manuals describing the use of R and Bioconductor for analysis of genomic datasets, available at `R and Bioconductor Manuals <http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual>`_.

  `Manuals <http://cran.r-project.org/manuals.html>`_ and contributed `documentation <http://cran.r-project.org/other-docs.html>`_ for R are available at the R-project.org website, and video tutorials are also available on Youtube; those posted by Tutorlol are brief, clear, and to the point.

  Materials from a series of mini-courses in R taught in 2010 at UCLA are available:
    + `Intro to programming and graphics <http://scc.stat.ucla.edu/page_attachments/0000/0141/10S-basicR.pdf>`_
    + `Data manipulation and functions <http://scc.stat.ucla.edu/page_attachments/0000/0143/S10_RProgII.pdf>`_
    + `Graphics for exploratory data analysis <http://scc.stat.ucla.edu/page_attachments/0000/0185/Graphics_course.pdf>`_
    + `Introductory statistics <http://scc.stat.ucla.edu/page_attachments/0000/0147/20100503_IntroStats.pdf>`_
    + `Linear regression <http://scc.stat.ucla.edu/page_attachments/0000/0188/reg_R_1_09S_slides.pdf>`_

  `A Little Book of R for Bioinformatics <http://a-little-book-of-r-for-bioinformatics.readthedocs.org/en/latest/>`_ is an on-line resource with information and exercises to provide practice in bioinformatics analysis of DNA sequences and other biological data in R. Many books on specific topics in R programming are also available through Amazon or other vendors.

+ Cloud computing resources
  + The case for cloud computing in genome informatics. Lincoln Stein, Genome Biol. 11(5):207, 2010 `Pubmed <http://www.ncbi.nlm.nih.gov/pubmed/20441614>`_
  + Galaxy Cloudman: delivering cloud compute clusters. Afgan et al, BMC Bioinformatics 11(Suppl 12):S4, 2010 `Full Text <http://www.biomedcentral.com/1471-2105/11/S12/S4>`_
  + `CloudBioLinux <http://cloudbiolinux.com/>`_ is an open-source project that provides a bioinformatics Linux system for cloud computing, pre-configured with a variety of software tools installed and ready to use.
  + A `tutorial <https://github.com/chapmanb/cloudbiolinux/blob/master/doc/intro/gettingStarted_CloudBioLinux.pdf?raw=true>`_ on getting started with CloudBioLinux on the Amazon Web Services Elastic Compute Cloud (EC2)
  + `Deploying Galaxy on the Cloud <http://userwww.service.emory.edu/%7Eeafgan/content/ppt/EnisAfgan_BOSC_2010.pdf>`_ slides from a presentation by Enis Afgan (Emory University) at the Bioinformatics Open Source Conference in Boston, July 2010
  + A `screencast <http://screencast.g2.bx.psu.edu/cloud/>`_ that provides a step-by-step guide to starting a Galaxy cluster in the EC2 environment
  + A `webpage <https://bitbucket.org/galaxy/galaxy-central/wiki/cloud>` that has the same information in text form, and is the basis for the screencast
  + The iPlant Collaborative, an NSF-funded project to create computational resources for plant biology research, provides access to cloud computing resources through `Atmosphere <http://www.iplantcollaborative.org/discover/atmosphere>`_
  + SeqWare Query Engine: storing and searching sequence data in the cloud. OConnor et al, BMC Bioinformatics 11(Suppl 12):S2, 2010 `Full Text <http://www.biomedcentral.com/1471-2105/11/S12/S2>`_
  + An overview of the Hadoop/MapReduce/HBase framework and its current applications in bioinformatics. Taylor, BMC Bioinformatics 11(Suppl 12):S1, 2010 `Full Text <http://www.biomedcentral.com/1471-2105/11/S12/S1>`_

+ Links to Linux command-line tutorials and resources
  Tutorials for AWK, a powerful tool for handling data tables
  + A set of `awk notes <http://people.bu.edu/scottm/AWK.NOTES>`_ from Boston University
  + Bruce Barnett's `awk tutorial <http://www.grymoire.com/Unix/Awk.html>`_
  + Greg Goebel's `awk tutorial <http://www.vectorsite.net/tsawk.html>`_
  + `Executing an awk command from R <http://teaching.software-carpentry.org/2013/01/16/1433/>`_ to simplify data exploratory analysis, from Lex Nederbragt
  Tutorials for bash shell scripting
  + A `tutorial <http://www.linuxconfig.org/bash-scripting-tutorial>`_ at linuxconfig.org
  + A `Getting Started With Bash <http://www.hypexr.org/bash_tutorial.php>`_ tutorial at hypexr.org
  + Mendel Cooper's `Advanced Bash Shell-Scripting Guide <http://tldp.org/LDP/abs/html/>`_
  Tutorials for sed, the command-line stream editor
  + A `tutorial <http://www.panix.com/%7Eelflord/unix/sed.html>`_ at Rutgers
  + Peter Krumins claims to have the `World's Best Introduction to Sed <http://www.catonmat.net/blog/worlds-best-introduction-to-sed/>`_; take a look and judge for yourself.
  + Bruce Barnett's `sed tutorial <http://www.grymoire.com/Unix/Sed.html>`_.

Links to other useful sites
***************************

+ The `SEQanswers <http://seqanswers.com/>`_ online community has forums on several topics related to sequencing; the bioinformatics forum is the most active.
+ The SEQanswers `Software Wiki <http://seqanswers.com/wiki/Software>`_ is a list of software for analysis of sequencing data
+ `Biostar <http://biostar.stackexchange.com/>`_ is another online community for questions and answers on bioinformatics and computational genomics.
+ Information on file formats used by the University of California - Santa Cruz Genome Browser is on the `FAQ list <http://genome.ucsc.edu/FAQ/FAQformat>`_
+ A manual for the Integrated Genome Browser visualization tool is `here <http://wiki.transvar.org/confluence/display/igbman/Home>`_
+ Course materials for a short course entitled `Introduction to R and Bioconductor <http://bioconductor.org/help/course-materials/2010/SeattleIntro/>`_, held in Seattle in Dec 2010
+ `Genomic Regions Enrichment of Annotations Tool <http://great.stanford.edu/>`_ - A web service to test for over-representation of specific ontology categories among genes near ChIP-seq peaks
+ Ben Langmead, author of several tools for sequence analysis, has made `course materials <https://github.com/BenLangmead/comp-genomics-class>`_ for a class in Computational Genomics available on Github.
+ An open-source book called `Introduction to Applied Bioinformatics <http://readiab.org/book/latest/>`_ has chapters on sequence alignment approaches and algorithms, for those interested in more detail about how that works.
+ `Next-gen-seq software <http://www.animalgenome.org/bioinfo/resources/nextgensoft.html>`_ - a list of software packages, both commercial and open-source, related to analysis of deep sequencing datasets
+ `Software <http://www.cbcb.umd.edu/software/>`_ from the Center for Bioinformatics and Computational Biology, University of Maryland - many useful programs, all open-source
+ `PLAZA <http://bioinformatics.psb.ugent.be/plaza/>`_: a comparative genomics resource to study gene and genome evolution in plants; described by Proost et al, Plant Cell 21:3718, 2010 `Full Text <http://www.plantcell.org/content/21/12/3718.full>`_
+ The European Bioinformatics Institute provides tools `ArrayExpressHTS and R-Cloud <http://www.ebi.ac.uk/Tools/rcloud/>`_ for analysis of transcriptome data


Last modified 18 December 2018.
Edits by `Ross Whetten <https://github.com/rwhetten>`_, `Will Kohlway <https://github.com/wkohlway>`_, & `Maria Adonay <https://github.com/amalgamaria>`_.
