
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Data Preprocessing and Quality Control &#8212; BIT 815 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/pyramid.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Neuton&amp;subset=latin" type="text/css" media="screen" charset="utf-8" />
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nobile:regular,italic,bold,bolditalic&amp;subset=latin" type="text/css" media="screen" charset="utf-8" />
<!--[if lte IE 6]>
<link rel="stylesheet" href="_static/ie6.css" type="text/css" media="screen" charset="utf-8" />
<![endif]-->

  </head><body>

    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="index.html">BIT 815 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <a class="reference external image-reference" href="http://www.ncsu.edu"><img alt="_images/NC_State.gif" src="_images/NC_State.gif" /></a>
<a class="reference external image-reference" href="http://www.cnr.ncsu.edu/fer/"><img alt="_images/FERlogo.png" src="_images/FERlogo.png" /></a>
<div class="section" id="data-preprocessing-and-quality-control">
<h1>Data Preprocessing and Quality Control<a class="headerlink" href="#data-preprocessing-and-quality-control" title="Permalink to this headline">¶</a></h1>
<div class="section" id="global-overview">
<h2>Global Overview<a class="headerlink" href="#global-overview" title="Permalink to this headline">¶</a></h2>
<p>Data from high-throughput DNA sequencing platforms (such Illumina, Ion Torrent, Pacific Biosciences, or Oxford Nanopore) can contain a variety of experimental artifacts and low-quality data, so quality control and data pre-processing are always a good idea. Common artifacts in sequences from Illumina and Ion Torrent instruments include copies of the adapter sequences, because these instruments use solid-phase PCR to amplify the template DNAs using in the sequencing reaction. FASTQ-format data contains both the DNA sequence data and an estimate of the probability of error for each base in the sequence; removing sequences (or regions of sequences) consisting of low-quality base calls is another common pre-processing step.</p>
</div>
<div class="section" id="objective">
<h2>Objective<a class="headerlink" href="#objective" title="Permalink to this headline">¶</a></h2>
<p>The objective of this session is to provide participants with experience in managing and processing DNA sequence data. A review of FASTQ format will be followed by practical exercises in data quality analysis with FastQC and data filtering with tools from the BBTools suite.</p>
</div>
<div class="section" id="description">
<h2>Description<a class="headerlink" href="#description" title="Permalink to this headline">¶</a></h2>
<p>The most common format for sequence data output from high-throughput instruments is FASTQ format (Cock et al., 2010), containing sequences coupled with their quality stored as text characters. The quality scores are -10 times the logarithm (to base 10) of the per-base probability of an incorrect nucleotide call at each position in the sequence, and are based on the PHRED quality score (Ewing and Green, 1998; Ewing et al., 1998). In the past, some instruments produced separate files containing DNA sequences (in Fasta format) and quality scores (<a href="#id1"><span class="problematic" id="id2">*</span></a>.qual files), so you may encounter these file formats as well.  The numerical quality scores (typically ranging from around zero to around 40) are converted into single characters to save space. A two-digit quality score must be followed by a space to distinguish it from the score for the next nucleotide, so one value requires three characters, while a single text character contains the same information and need not be separated by a space from the next single character.</p>
<p>Unfortunately, multiple different conventions have been used for encoding quality data in text characters using the American Standard Code for Information Interchange (ASCII), and knowing which convention was used for any given dataset is important. The Wikipedia page on <a class="reference external" href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ format</a> is the most up-to-date source of information about the different encoding schemes used in the past and the current encoding used by Illumina instruments (called Illumina 1.8+ in the figure).</p>
<p>Several tools for summarizing the frequency and identity of contaminating adapter sequences and the distribution of quality scores per base or per sequence are available; we will use the Java program FastQC for data quality summaries. Many different software tools exist for removing low-quality base calls from the 5’ end or 3’ ends of sequence reads, for trimming off sequencing adaptor sequences or other extraneous sequences, and for separating reads from a single dataset into multiple datasets based on the presence of “barcode” sequences at the beginning of the sequence reads. We will use both fastx-toolkit (a simple, lightweight but limited toolset) and BBTools (a Java-based and extremely comprehensive package of programs) for trimming and filtering reads.</p>
</div>
<div class="section" id="key-facts">
<h2>Key Facts<a class="headerlink" href="#key-facts" title="Permalink to this headline">¶</a></h2>
<p>DNA sequence data as produced from the instrument can vary in quality and read length, and some exploratory analysis and filtering of the data is essential before beginning the process of analyzing the sequence to achieve  experimental objectives. Understanding the format in which sequence data are provided is essential, as there are (unfortunately) multiple versions of the “standard” FASTQ format in use.  Converting FASTQ format to FASTA is simply a matter of stripping out the quality data and changing the character at the beginning of the sequence header line, but converting FASTA and quality score files into a single FASTQ format file requires converting numerical quality scores into characters using the ASCII lookup table.</p>
</div>
<div class="section" id="exercises">
<h2>Exercises<a class="headerlink" href="#exercises" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li>Sequence format conversion tools. The key difference between FASTA and FASTQ format is that FASTQ contains quality scores. If you have a FASTA file containing sequences, and no accompanying file of quality scores, there is no meaningful way to provide real quality scores for creation of a FASTQ file.  If you have a separate .qual file of quality scores, you can use various tools to merge the FASTA sequences with the quality scores to produce a FASTQ format file. Search online for a user guide to the BBTools package of programs, which is installed on the VCL machine image and has a reformat.sh script that can perform this conversion as well as many others. Read through the help information to learn about the capabilities of the BBTools package of programs, or see Bushnell et al (2017) for more details on the BBMerge tool from this package that can merge paired-end reads into longer single sequences.</li>
<li>Software: fastx-toolkit and FastQC. FastQC is a Java program that will run on Windows, Mac, or Linux, and is already installed on the VCL machine image - the link here is provided for those interested in installing the software on another machine. The fastx-toolkit package of programs is available in the Ubuntu repository, and is also already installed on the machine image.  These programs differ in that FastQC has a nice graphic interface and produces pretty graphs describing various aspects of data quality, but has no capability to filter, trim, or otherwise modify sequence files to address problems made apparent in those pretty graphs. The fastx-toolkit package can also produce graphs, although they aren’t quite as pretty, but it does have</li>
<li>Using fastx-toolkit for exploratory analysis. The first exercise will use fastx-toolkit to analyze and process the c3.fq.gz sequence file from the /data/AtRNAseq directory on your VCL machine instance. The fastx-toolkit programs are simple and somewhat dated, but still useful. Download the QCexercises.sh shell script with exercises by executing the following command at a terminal prompt:</li>
</ol>
<blockquote>
<div>wget <a class="reference external" href="http://www4.ncsu.edu/~rosswhet/BIT815/resources/QCexercises.sh">http://www4.ncsu.edu/~rosswhet/BIT815/resources/QCexercises.sh</a></div></blockquote>
<ol class="arabic simple" start="4">
<li>Run the script using the command bash QCexercises.sh from a terminal prompt. The shell script should create two image files (with a .png extension) in the directory where you run it - use one of the image viewer programs available under Graphics in the Application menu of the Linux system to view these image files, or just use the display command at a terminal prompt.</li>
</ol>
<blockquote>
<div>display c3.qual.png &amp;</div></blockquote>
<ol class="arabic simple" start="5">
<li>Open the QCexercises.sh file with the Geany programming text editor (listed under the Development item in the Applications menu)  and read the code and comments to learn what the script does. You can run the script multiple times with different input sequence files, changing the name of the file to be used as input to the series of programs, or you can simply copy the commands from the script file to a terminal window and run each command from the command line, one step at a time.</li>
<li>Run FastQC on the same file, and compare the results. FastQC produces a report containing an HTML file that can be opened using a browser. By default FastQC tries to write the output to a directory in the same place where the sequence file is found - this won’t work on your virtual machine image, because the sequence files are in the /data directory and ordinary users don’t have write permission in that directory. You can specify an alternative place to save the output using the -o option on the command line:</li>
</ol>
<blockquote>
<div>fastqc -o ./ /data/AtRNAseq/c3.fq.gz</div></blockquote>
<p>The resulting HTML output file can be opened with a browser from the command line; e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">palemoon</span> <span class="n">c3_fastqc</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<ol class="arabic simple" start="7">
<li>Using fastx-toolkit to remove adaptor sequences and trim off low-quality bases. Find the web page with instructions on the individual programs in the fastx-toolkit package (remember, the information is out there - you just have to find it). Use the fastx_clipper and fastq_quality_filter programs to remove any copies of the Illumina sequencing adaptor (GATCGGAAGAGCTCGTATGCCGTCTTCTGCTTGAAA) and to trim low-quality bases so that 80% of the bases in each read have a quality score of 15 or higher. Note that the fastx-toolkit programs accept input from STDIN and deliver output to STDOUT by default, so they are designed to be linked together in pipelines for processing data without the need to save intermediate files. See the QCexercises.sh script file for an example of how to format the command to send decompressed fastq-format sequence to fastx-toolkit commands.</li>
</ol>
<p><strong>:red:`NOTE: the -Q33 option is required`</strong> - fastx-toolkit by default assumes that fastq-format sequences have quality scores from an older version of the Illumina base-calling software, and if you don’t tell it to use the new version, it will return an error message.</p>
<ol class="arabic simple" start="8">
<li><span class="underline">Using BBTools programs to remove adaptor sequences and trim low-quality bases.</span> The BBTools programs are installed in the /usr/local/bbmap directory of the Linux system, and this directory has been included in the $PATH environment variable, so you can run them by typing the name of the command at a terminal prompt, for example bbduk.sh to run the bbduk.sh program. Executing this command with no arguments will print a user guide for the command to the terminal screen, so this is one way to learn what options and arguments each command accepts. A web search will lead you to a BBTools User Guide at the DOE Joint Genome Institute, because the author (Brian Bushnell) is a bioinformatics specialist at JGI. NOTE: many of the BBTools programs are Java-based, so they can  be used on any operating system that has Java installed, but you can read the user guides for all the commands without installing Java. By default, the bbduk.sh and bbduk2.sh programs do not use the same sliding window approach for quality trimming as does fastq_quality_filter, but setting the appropriate options during execution of either bbduk.sh or bbduk2.sh will allow that approach to be used. For more information about alternative ways of quality trimming, see this <a class="reference external" href="http://seqanswers.com/forums/showthread.php?t=42776&amp;page=7">SeqAnswers Forum</a> thread, and look for post #134.</li>
<li><span class="underline">Summarizing sequence data characteristics using FastQC.</span> You can run FastQC either from the command line, providing the names of sequence files to be processed as arguments, or from a graphic user interface. Typing the  command <code class="code docutils literal notranslate"><span class="pre">fastqc</span></code> without providing an input filename will start the program in interactive mode, where you choose which file to analyze from the File menu, while providing a file “glob” using wildcard characters will run the program on every sequence file that matches the filename pattern, e.g.:</li>
</ol>
<blockquote>
<div>fastqc /data/AtRNAseq/[ct][123].fq.gz</div></blockquote>
<p>Note that the FastQC program can process gzip-compressed sequence files without saving an uncompressed version - this is important for saving disk space when hundreds of gigabytes of compressed sequence files need to be processed.</p>
</div>
<div class="section" id="additional-resources">
<h2>Additional Resources<a class="headerlink" href="#additional-resources" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Wikipedia has information on <a class="reference external" href="http://en.wikipedia.org/wiki/Fasta_format">FASTA</a> and <a class="reference external" href="http://en.wikipedia.org/wiki/Fastq">FASTQ</a> sequence formats.</li>
<li>The University of California - Santa Cruz Genome Browser site maintains a <a class="reference external" href="http://genome.ucsc.edu/FAQ/FAQformat.html">FAQ</a> with information about many different file formats used in analysis of deep sequencing data</li>
<li>The fastx-toolkit <a class="reference external" href="http://hannonlab.cshl.edu/fastx_toolkit/commandline.html">webpage</a> has information about the fastx-toolkit package of programs for quality control and manipulation of FASTA and FASTQ files.</li>
<li>The FastQC <a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc">webpage</a> has information about the FastQC program, and details on FastQC output are provided in the <a class="reference external" href="http://www4.ncsu.edu/~rosswhet/BIT815/Overview/Week2/FastQC_details.pdf">FastQC_details.pdf</a> document.</li>
<li>Another program suitable for adapter trimming is called “flexbar” - this program can also split reads into different files based on the presence of specific “barcode” sequences detected in the sequence reads. Such barcodes are common in GBS and RAD-seq applications, and the ability to detect variable-length barcodes is somewhat unusual. The manual for flexbar is on <a class="reference external" href="http://sourceforge.net/p/flexbar/wiki/Manual/">Sourceforge</a>, and the <a class="reference external" href="http://www.mdpi.com/2079-7737/1/3/895">publication</a> describing the software is also available.</li>
<li>The BBtools suite of programs was announced on the SeqAnswers forum, and the correspondence between the program developer and users is archived as a resource for others to learn how to use the various tools in the suite. The announcements and correspondence are in separate threads for individual programs; the <a class="reference external" href="http://seqanswers.com/forums/tags.php?tag=bbtools">list of tagged posts</a> can be viewed to see links to the individual threads. The software is available at the Sourceforge <a class="reference external" href="https://sourceforge.net/projects/bbmap/">project page</a>.</li>
<li>Breese MR, Liu Y. (2013) NGSUtils: a software suite for analyzing and manipulating next-generation sequencing datasets. Bioinformatics 29: 494-496, 2013. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/23314324">PMID 23314324</a> (<strong>*Note</strong>: This paper describes a set of software tools for managing the process of data QC and format conversion, including tools for filtering datasets of paired-end reads to find single reads where the paired-end read was removed by a quality-filtering step*).</li>
<li>Cock PJ, Fields CJ, Goto N, Heuer ML, and Rice PM. (2010) The Sanger FASTQ file format for sequences with quality scores, and the Solexa/Illumina FASTQ variants. Nucleic Acids Res. 38: 1767–1771. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20015970">PMID 20015970</a> (<strong>*Note</strong>: This is the only formal publication I know of that describes the different versions of the FASTQ sequence format, and it is not as up-to-date as the Wikipedia page on FASTQ format*).</li>
<li>Ewing B, Hillier L, Wendl MC, Green P (1998). Base-calling of automated sequencer traces using phred. I. Accuracy assessment. Genome Res. 8 (3): 175–185. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/9521921">PMID 9521921</a></li>
<li>Ewing B, Green P (1998). Base-calling of automated sequencer traces using phred. II. Error probabilities. Genome Res. 8 (3): 186–194. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/9521922">PMID 9521922</a></li>
</ul>
<p>Last modified 18 December 2018.
Edits by <a class="reference external" href="https://github.com/rwhetten">Ross Whetten</a>, <a class="reference external" href="https://github.com/wkohlway">Will Kohlway</a>, &amp; <a class="reference external" href="https://github.com/amalgamaria">Maria Adonay</a>.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Data Preprocessing and Quality Control</a><ul>
<li><a class="reference internal" href="#global-overview">Global Overview</a></li>
<li><a class="reference internal" href="#objective">Objective</a></li>
<li><a class="reference internal" href="#description">Description</a></li>
<li><a class="reference internal" href="#key-facts">Key Facts</a></li>
<li><a class="reference internal" href="#exercises">Exercises</a></li>
<li><a class="reference internal" href="#additional-resources">Additional Resources</a></li>
</ul>
</li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/data-preprocessing-qc.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="index.html">BIT 815 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Ross Whetten.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.2.
    </div>
  </body>
</html>