
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>BIT 815, Deep Sequencing Data Analysis &#8212; BIT 815 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Semester Overview, 2019" href="course-schedule.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <a class="reference external image-reference" href="http://www.ncsu.edu"><img alt="_images/NC_State2.gif" src="_images/NC_State2.gif" /></a>
<a class="reference external image-reference" href="http://www.cnr.ncsu.edu/fer/"><img alt="_images/FERlogo2.png" src="_images/FERlogo2.png" /></a>
<div class="section" id="bit-815-deep-sequencing-data-analysis">
<h1>BIT 815, Deep Sequencing Data Analysis<a class="headerlink" href="#bit-815-deep-sequencing-data-analysis" title="Permalink to this headline">¶</a></h1>
<div class="section" id="instructor-dr-ross-whetten">
<h2>Instructor: Dr. Ross Whetten<a class="headerlink" href="#instructor-dr-ross-whetten" title="Permalink to this headline">¶</a></h2>
<div class="section" id="course-description">
<h3>Course Description<a class="headerlink" href="#course-description" title="Permalink to this headline">¶</a></h3>
<p>The course BIT815, Analysis of Deep Sequencing Data, is designed to introduce biologists to the Linux command-line computing environment, to cloud computing, and to open-source software for analysis of next-generation sequencing data.Class sessions consist of two-hour blocks, each beginning with presentation and discussion of a specific topic, followed by hands-on cloud computing exercises using model datasets. A total of 45 two-hour blocks are scheduled over a 15-week period, and the course is offered once per calendar year. The importance of cloud computing is emphasized, due to the increasing demands for RAM and storage space required for analysis and storage of high-throughput DNA sequencing data, and the cost-effectiveness and flexibility provided by cloud computing solutions.Applications of sequencing discussed include genome sequencing (both de-novo and resequencing), transcriptome analysis, discovery of sequence and structural variations, ChIP-seq methods for mapping DNA-protein interactions, and genotyping by sequencing (GBS and RAD-seq methods). For each application of sequencing technology, discussion topics include experimental design strategies, methods for library construction, sources of experimental and biological variation, and analytical approaches available in open-source software packages. Computing exercises utilize the software discussed, and provide participants with the opportunity to carry out analysis of sample datasets using a virtual machine image through the NC State University Virtual Computing Lab. This Linux system is customized to provide the bioinformatics software described during the course, and is available for class participants to use at any time. The objective of the course is not to make course participants experts in every aspect of sequence analysis, but instead to empower participants to learn the specific skills they need by teaching basic skills in command-line Linux computing, and providing an introduction to the literature and on-line resources. The course is directed at graduate students, but has also attracted participation from faculty, post-doctoral researchers, and research technicians interested in expanding their skills in the area of sequence data analysis.</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="course-schedule.html">Semester Overview, 2019</a><ul>
<li class="toctree-l2"><a class="reference internal" href="course-schedule.html#course-schedule">Course Schedule</a></li>
<li class="toctree-l2"><a class="reference internal" href="course-schedule.html#general-background-information-and-course-resources">General background information and course resources</a></li>
<li class="toctree-l2"><a class="reference internal" href="course-schedule.html#a-flow-chart-overview-of-dna-sequencing-experiments">A flow-chart overview of DNA sequencing experiments</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="topics.html">Topics</a><ul>
<li class="toctree-l2"><a class="reference internal" href="topics.html#id1">Sequencing Instruments</a></li>
<li class="toctree-l2"><a class="reference internal" href="topics.html#id2">Experimental Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="topics.html#id3">Data Preprocessing and QC</a></li>
<li class="toctree-l2"><a class="reference internal" href="topics.html#id4">Transcriptome Assembly</a></li>
<li class="toctree-l2"><a class="reference internal" href="topics.html#id5">Genome Assembly</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="readings.html">Useful Publications and Websites</a><ul>
<li class="toctree-l2"><a class="reference internal" href="readings.html#global-overview-books-and-papers">Global overview books and papers</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#library-construction-and-experimental-design">Library construction and experimental design</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#data-formats-data-management-and-alignment-software-tools">Data formats, data management, and alignment software tools</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#data-quality-assessment-filtering-and-correction">Data quality assessment, filtering, and correction</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#de-novo-assembly">De novo assembly</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#chromatin-analysis">Chromatin analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#transcriptome-analysis">Transcriptome analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#comparing-genomes-and-assemblies-variant-detection">Comparing genomes and assemblies; variant detection</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#population-genomics">Population Genomics</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#workspace-environments">Workspace environments</a></li>
<li class="toctree-l2"><a class="reference internal" href="readings.html#links-to-other-useful-sites">Links to other useful sites</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="HPC.html">Using the HPC for Bioinformatics Computing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="HPC.html#global-overview">Global overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="HPC.html#job-submission-and-management">Job submission and management</a></li>
<li class="toctree-l2"><a class="reference internal" href="HPC.html#available-software">Available software</a></li>
<li class="toctree-l2"><a class="reference internal" href="HPC.html#setting-up-a-job-submission-script">Setting up a job submission script</a></li>
</ul>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></li>
<li><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></li>
<li><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></li>
</ul>
<p>Last modified 18 December 2018.
Edits by <a class="reference external" href="https://github.com/rwhetten">Ross Whetten</a>, <a class="reference external" href="https://github.com/wkohlway">Will Kohlway</a>, &amp; <a class="reference external" href="https://github.com/amalgamaria">Maria Adonay</a>.</p>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="#">BIT 815</a></h1>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="course-schedule.html">Semester Overview, 2019</a></li>
<li class="toctree-l1"><a class="reference internal" href="topics.html">Topics</a></li>
<li class="toctree-l1"><a class="reference internal" href="readings.html">Useful Publications and Websites</a></li>
<li class="toctree-l1"><a class="reference internal" href="HPC.html">Using the HPC for Bioinformatics Computing</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
      <li>Next: <a href="course-schedule.html" title="next chapter">Semester Overview, 2019</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Ross Whetten.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>