
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Useful Publications and Websites &#8212; BIT 815 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/pyramid.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Using the HPC for Bioinformatics Computing" href="HPC.html" />
    <link rel="prev" title="Topics" href="topics.html" />
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Neuton&amp;subset=latin" type="text/css" media="screen" charset="utf-8" />
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nobile:regular,italic,bold,bolditalic&amp;subset=latin" type="text/css" media="screen" charset="utf-8" />
<!--[if lte IE 6]>
<link rel="stylesheet" href="_static/ie6.css" type="text/css" media="screen" charset="utf-8" />
<![endif]-->

  </head><body>

    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="HPC.html" title="Using the HPC for Bioinformatics Computing"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="topics.html" title="Topics"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BIT 815 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <a class="reference external image-reference" href="http://www.ncsu.edu"><img alt="_images/NC_State.gif" src="_images/NC_State.gif" /></a>
<a class="reference external image-reference" href="http://www.cnr.ncsu.edu/fer/"><img alt="_images/FERlogo.png" src="_images/FERlogo.png" /></a>
<div class="section" id="useful-publications-and-websites">
<h1>Useful Publications and Websites<a class="headerlink" href="#useful-publications-and-websites" title="Permalink to this headline">¶</a></h1>
<div class="section" id="global-overview-books-and-papers">
<h2>Global overview books and papers<a class="headerlink" href="#global-overview-books-and-papers" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>The Biostar Handbook - Bioinformatics Data Analysis Guide. Istvan Albert and others. Available <a class="reference external" href="https://biostar.myshopify.com/">online</a></li>
<li>Next generation quantitative genetics in plants. Jiménez-Gómez, Frontiers in Plant Science 2:77, 2011 <a class="reference external" href="http://www.frontiersin.org/Plant_Physiology/10.3389/fpls.2011.00077/full">Full Text</a> [<em>Equally relevant to animal and microbial systems</em>]</li>
<li>Sense from sequence reads: methods for alignment and assembly. Flicek &amp; Birney, Nat Methods 6(11 Suppl):S6-S12, 2009. <a class="reference external" href="http://www.nature.com/nmeth/journal/v6/n11s/full/nmeth.1376.html">Full Text</a></li>
</ul>
</div>
<div class="section" id="library-construction-and-experimental-design">
<h2>Library construction and experimental design<a class="headerlink" href="#library-construction-and-experimental-design" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Statistical design and analysis of RNA sequencing data. Auer &amp; Doerge, Genetics 185(2):405-16, 2010. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2881125">PubMedCentral</a></li>
<li>Biases in Illumina transcriptome sequencing caused by random hexamer priming. Hansen et al., Nucleic Acids Res. 38(12): e131, 2010. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2896536">PubMedCentral</a></li>
<li>Analyzing and minimizing PCR amplification bias in Illumina sequencing libraries. Aird et al, Genome Biology 12:R18, 2011 <a class="reference external" href="http://genomebiology.com/2011/12/2/R18">Full Text</a></li>
<li>Amplification-free Illumina sequencing-library preparation facilitates improved mapping and assembly of GC-biased genomes. Kozarewa et al, Nature Methods 6(4):291-295, 2009 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2664327/">PubMedCentral</a></li>
<li>Cost-effective, high-throughput DNA sequencing libraries for multiplexed target capture. Rohland &amp; Reich, Genome Research 22(5): 939–946, 2012. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3337438/">PubMedCentral</a></li>
<li>Predicting the molecular complexity of sequencing libraries. Daley &amp; Smith, Nature Methods 10(4):325-327, 2013 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3612374/">PubMedCentral</a></li>
<li>RNA-seq differential expression studies: more sequence or more replication? Liu et al., Bioinformatics 30: 301 - 304, 2014. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/30/3/301.long">Publisher Web Site</a></li>
<li>Power analysis and sample size estimation for RNA-seq differential expression. Ching et al., RNA 20: 1684 - 1696, 2014. <a class="reference external" href="http://rnajournal.cshlp.org/content/20/11/1684.short">Publisher Web Site</a></li>
<li>Guidance for RNA-seq co-expression network construction and analysis: safety in numbers. Ballouz et al., Bioinformatics 31: 2123 - 2130, 2014. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/31/13/2123.short">Publisher Web Site</a></li>
<li>Points of significance: replication. Blainey et al., Nature Methods 11: 879–880, 2014. <a class="reference external" href="http://www.nature.com/nmeth/journal/v11/n9/full/nmeth.3091.html">Publisher Web Site</a></li>
<li>Points of Significance: Nested designs. Krzywinski et al., Nature Methods 11: 977–978, 2014 <a class="reference external" href="http://www.nature.com/nmeth/journal/v11/n10/full/nmeth.3137.html">Publisher Web Site</a></li>
<li>Points of significance: Sources of variation Altman &amp; Krzywinski. Nature Methods 12: 5 – 6, 2015 <a class="reference external" href="http://www.nature.com/nmeth/journal/v12/n1/full/nmeth.3224.html">Publisher Web Site</a></li>
</ul>
</div>
<div class="section" id="data-formats-data-management-and-alignment-software-tools">
<h2>Data formats, data management, and alignment software tools<a class="headerlink" href="#data-formats-data-management-and-alignment-software-tools" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>The Sequence Alignment/Map format and SAMtools. Li et al, Bioinformatics 25(16):2078-9, 2009 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2723002">PubMedCentral</a></li>
<li>SAM format specification <a class="reference external" href="http://samtools.sourceforge.net/SAM1.pdf">file</a></li>
<li>Efficient storage of high throughput sequencing data using reference-based compression. Fritz et al, Genome Res 21(5):734-40, 2011. <a class="reference external" href="http://genome.cshlp.org/content/21/5/734.long">Full Text</a></li>
<li>Compression of DNA sequence reads in FASTQ format. Deorowicz &amp; Grabowski, Bioinformatics 27(6):860-2, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21252073">PubMed</a></li>
<li>Fast and accurate short read alignment with Burrows-Wheeler transform. Li &amp; Durbin, Bioinformatics 25(14):1754-60, 2009. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2705234">PubMedCentral</a></li>
<li>Improving SNP discovery by base alignment quality. Li H, Bioinformatics 27(8):1157-8, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21320865">PubMed</a></li>
<li>BEDTools: a flexible suite of utilities for comparing genomic features. Quinlan and Hall, Bioinformatics 26:841-842, 2010. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/26/6/841.full.pdf+html">Publisher Website</a></li>
</ul>
</div>
<div class="section" id="data-quality-assessment-filtering-and-correction">
<h2>Data quality assessment, filtering, and correction<a class="headerlink" href="#data-quality-assessment-filtering-and-correction" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>HTQC: a fast quality control toolkit for Illumina sequencing data. Yang et al, BMC Bioinformatics 14:33, 2013. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3571943/">PubMed</a></li>
<li>FastQC: a quality control tool for high-throughput sequence data. <a class="reference external" href="http://www.bioinformatics.bbsrc.ac.uk/projects/fastqc/">Home Page</a></li>
<li>FASTX-toolkit: FASTQ/A short-reads pre-processing tools <a class="reference external" href="http://hannonlab.cshl.edu/fastx_toolkit/">Home Page</a></li>
<li>QuorUM: an error corrector for Illumina reads.  Marçais et al. <a class="reference external" href="http://arxiv.org/abs/1307.3515">2013 Arxiv preprint</a> or <a class="reference external" href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0130821">2015 PLoSOne paper</a></li>
<li>Quake: quality-aware detection and correction of sequencing errors. Kelley et al, Genome Biol 11(11):R116, 2010. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21114842">PubMed</a></li>
<li>Reference-free validation of short read data. Schröder et al, PLoS One 5(9):e12681, 2010. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2943903">PubMedCentral</a></li>
<li>Correction of sequencing errors in a mixed set of reads. Salmela, Bioinformatics 26(10):1284, 2010. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/26/10/1284.long">Full Text</a> [<em>Includes error correction of SOLiD reads in colorspace.</em>]</li>
<li>Repeat-aware modeling and correction of short read errors. Yang et al, BMC Bioinformatics 12(Supp1):S52, 2011 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3044310">PubMedCentral</a> [<em>Requires a reference sequence.</em>]</li>
<li>HiTEC: accurate error correction in high-throughput sequencing data. Ilie et al, Bioinformatics 27(3):295, 2011 <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/27/3/295.long">Full Text</a></li>
<li>Error correction of high-throughput sequencing datasets with non-uniform coverage. Medvedev et al., Bioinformatics 27(13):i137-41, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3117386">PubMedCentral</a></li>
<li>Characterization of the Conus bullatus genome and its venom-duct transcriptome. Hu et al., BMC Genomics 12:60, 2011 <a class="reference external" href="http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-12-60">Full Text</a> [<em>Includes a novel strategy for estimating genome size from a partial transcriptome assembly and low-coverage (3x) genome sequence.</em>]</li>
</ul>
</div>
<div class="section" id="de-novo-assembly">
<h2>De novo assembly<a class="headerlink" href="#de-novo-assembly" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Velvet: algorithms for de novo short read assembly using de Bruijn graphs. Zerbino &amp; Birney, Genome Res 18(5):821-9, 2008. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336801">PubMedCentral</a></li>
<li>Assembly of large genomes using second-generation sequencing. Schatz et al, Genome Res 20(9):1165-73, 2010. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928494">PubMedCentral</a></li>
<li>High-quality draft assemblies of mammalian genomes from massively parallel sequence data. Gnerre et al, PNAS 108(4): 1513-18, 2011 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3029755">PubMedCentral</a></li>
<li>Genome assembly has a major impact on gene content: a comparison of annotation in two Bos taurus assemblies. Florea  et al., PLoS One 6(6):e21400, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3120881/">PubMedCentral</a></li>
<li>Artemis: an integrated platform for visualization and analysis of high-throughput sequence-based experimental data. Carver et al, Bioinformatics 28(4):464 - 469, 2012 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3278759/">PubMedCentral</a></li>
<li>Efficient de novo assembly of large genomes using compressed data structures. Simpson &amp; Durbin, Genome Research 22:549-556, 2012 <a class="reference external" href="http://genome.cshlp.org/content/22/3/549.full">Full Text</a> [<em>Describes the String Graph Assembler (SGA), which assembled a human genome in less than 6 days using 54 Gb of RAM and a 123-processor compute cluster for calculation of an FM-index of the 1.2 billion reads</em>]</li>
<li>Readjoiner: a fast and memory efficient string graph-based sequence assembler. Gonnella &amp; Kurtz, BMC Bioinformatics 13: 82, 2012 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3507659">PubMedCentral</a></li>
<li>Assemblathon 1: A competitive assessment of de novo short read assembly methods. Earl et al, Genome Research 21:2224-2241, 2011 <a class="reference external" href="http://genome.cshlp.org/content/early/2011/09/16/gr.126599.111.full.pdf+html">Full Text</a></li>
</ul>
</div>
<div class="section" id="chromatin-analysis">
<h2>Chromatin analysis<a class="headerlink" href="#chromatin-analysis" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Bias Correction
+ Identifying and mitigating bias in next-generation sequencing methods for chromatin biology. Meyer and Liu, Nat Rev Genetics 15: 709 - 721, 2014 <a class="reference external" href="http://www.nature.com/nrg/journal/v15/n11/abs/nrg3788.html">Publisher Web Site</a></li>
<li>Chromatin Immunoprecipitation sequencing: ChIP-seq
+ ChIP-seq: advantages and challenges of a maturing technology. Park, Nat Rev Genet. 10:669-80, 2009 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3191340/">PubMed</a>
+ ChIP-seq and Beyond: new and improved methodologies to detect and characterize protein-DNA interactions. Furey, Nat Rev Genet 13: 840–852, 2012 <a class="reference external" href="http://www.nature.com/nrg/journal/v13/n12/full/nrg3306.html">Publisher Web Site</a>
+ MuMoD: a Bayesian approach to detect multiple modes of protein–DNA binding from genome-wide ChIP data. Narlikar, Nucleic Acids Res 41:21–32, 2013 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3592440/">PubMed</a></li>
<li>Chromatin conformation
+ A decade of 3C technologies: insights into nuclear organization. de Wit &amp; de Laat, Genes &amp; Devel  26: 11-24, 2012 <a class="reference external" href="http://genesdev.cshlp.org/content/26/1/11.full">Publisher Website</a>
+ Exploring the three-dimensional organization of genomes: interpreting chromatin interaction data. Dekker et al, Nature Reviews Genetics 14: 390–403, 2013 <a class="reference external" href="http://www.nature.com/nrg/journal/v14/n6/full/nrg3454.html">Publisher Website</a></li>
</ul>
</div>
<div class="section" id="transcriptome-analysis">
<h2>Transcriptome analysis<a class="headerlink" href="#transcriptome-analysis" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>General considerations for RNA-seq library construction
+ Molecular indexing enables quantitative targeted RNA sequencing and reveals poor efficiencies in standard library preparations. Fu et al, PNAS 111:1891–1896, 2014 <a class="reference external" href="http://www.pnas.org/content/111/5/1891">Publisher Web Site</a></li>
<li>Assembly and comparison to genome
+ A glance at quality score: implication for de novo transcriptome reconstruction of Illumina reads. Mbandi et al., Frontiers in Genetics 2014. <a class="reference external" href="http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00017/full?">Publisher Website</a>
+ Full-length transcriptome assembly from RNA-Seq data without a reference genome. Grabherr et al, Nature Biotechnology 29:644 - 652, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21572440">PubMed</a> [<em>Software called `Trinity &lt;http://trinityrnaseq.sourceforge.net/&gt;`_; available on Sourceforge.</em>]
+ Comprehensive analysis of RNA-Seq data reveals extensive RNA editing in a human transcriptome. Peng et al, Nature Biotechnology 30:253 - 260, 2012. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22327324">PubMed</a> [<em>Several comments on this paper question whether the reported differences are in fact evidence of editing or are simply sequencing errors - the authors stand by their conclusions, but the controversy demonstrates the importance of robust data analysis methods.</em>]
+ Optimization of de novo transcriptome assembly from next-generation sequencing data. Surget-Groba &amp; Montoya-Burgos, Genome Res 20(10):1432-40, 2010. <a class="reference external" href="http://genome.cshlp.org/content/20/10/1432.long">Full Text</a>
+ Rnnotator: an automated de novo transcriptome assembly pipeline from stranded RNA-Seq reads. Martin et al, BMC Genomics 11:663, 2010 <a class="reference external" href="http://www.biomedcentral.com/1471-2164/11/663">Full Text</a>
+ De novo assembly and analysis of RNA-seq data. Robertson et al, Nature Methods 7:909-912, 2010 <a class="reference external" href="http://www.nature.com/nmeth/journal/v7/n11/full/nmeth.1517.html">Full Text</a> [<em>Describes Trans-ABySS, a pipeline to use the ABySS parallel assembler for de novo transcriptome analysis.</em>]</li>
<li>Differential expression analysis
+ Robust adjustment of sequence tag abundance. Baumann &amp; Doerge, Bioinformatics 2013 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/24108185">PubMed</a>
+ R-SAP: a multi-threading computational pipeline for the characterization of high-throughput RNA-sequencing data. Mittal &amp; McDonald, Nucleic Acids Res, 2012 <a class="reference external" href="http://nar.oxfordjournals.org/content/early/2012/01/28/nar.gks047.long">Full Text</a>
+ Targeted RNA sequencing reveals the deep complexity of the human transcriptome. Mercer et al, Nature Biotechnology 30:99 - 104, 2012 <a class="reference external" href="http://www.nature.com/nbt/journal/v30/n1/full/nbt.2024.html">Publisher Website</a>
+ Differential gene and transcript expression analysis of RNA-Seq experiments with TopHat and Cufflinks. Trapnell et al, Nature Protocols 7:562 - 578, 2012 <a class="reference external" href="http://www.nature.com/nprot/journal/v7/n3/full/nprot.2012.016.html">Publisher Website</a>
+ Characterization and improvement of RNA-Seq precision in quantitative transcript expression profiling. Łabaj et al, Bioinformatics 27:i383 - i391, 2011 <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/27/13/i383.full.pdf+html">Full Text</a>
+ Improving RNA-Seq expression estimates by correcting for fragment bias. Roberts et al, Genome Biol 12:R22, 2011 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3129672/">PubMed Central</a>
+ Cloud-scale RNA-sequencing differential expression analysis with Myrna. Langmead et al, Genome Biol 11:R83, 2010 <a class="reference external" href="http://genomebiology.com/2010/11/8/R83">Full Text</a>
+ From RNA-seq reads to differential expression results. Oshlack et al, Genome Biol 11(12):220, 2010 <a class="reference external" href="http://genomebiology.com/content/11/12/220">Full Text</a>
+ DEGseq: an R package for identifying differentially expressed genes from RNA-seq data. Wang et al., Bioinformatics. 26(1):136-8. 2010 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/19855105">PubMed</a>
+ DEseq: Differential expression analysis for sequence count data. Anders and Huber, Genome Biology 11:R106, 2010 <a class="reference external" href="http://genomebiology.com/2010/11/10/R106">Full Text</a>
+ Moderated estimation of fold change and dispersion for RNA-Seq data with DESeq2. Love et al, BioRxiv doi: 10.1101/002832, 2014 <a class="reference external" href="http://biorxiv.org/content/early/2014/02/19/002832">Full Text</a>
+ edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Robinson et al., Bioinformatics 26(1):139-40 2010 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2796818">PubMedCentral</a>
+ Two-stage Poisson model for testing RNA-seq data. Auer and Doerge, SAGMB 10(1), article 26 <a class="reference external" href="http://www.bepress.com/sagmb/vol10/iss1/art26/">Full Text</a>
+ Experimental design, preprocessing, normalization and differential expression analysis of small RNA sequencing experiments. McCormick et al., Silence2(1):2, 2011 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3055805">PubMedCentral</a>
+ RNA-Seq gene expression estimation with read mapping uncertainty. Li et al, Bioinformatics 26:493-500, 2010 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2820677">PubMedCentral</a> [<em>Describes the RSEM software package.</em>]</li>
</ul>
</div>
<div class="section" id="comparing-genomes-and-assemblies-variant-detection">
<h2>Comparing genomes and assemblies; variant detection<a class="headerlink" href="#comparing-genomes-and-assemblies-variant-detection" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><ul class="simple">
<li>Toward better understanding of artifacts in variant calling from high-coverage samples. Heng Li, Bioinformatics 30, 2843, 2014 <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4271055/">PubMedCentral</a></li>
<li>Versatile and open software for comparing large genomes. Kurtz et al, Genome Biol (5(2):R12, 2004. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC395750">PubMedCentral</a> [<em>Describes the MUMmer software for full-genome alignment &amp; comparisons.</em>]</li>
<li>Searching for SNPs with cloud computing. Langmead et al, Genome Biol 10(11):R134, 2009 <a class="reference external" href="http://genomebiology.com/content/10/11/R134">Full Text</a></li>
<li>Calling SNPs without a reference sequence. Ratan et al, BMC Bioinformatics 11:130, 2010 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2851604">PubMedCentral</a></li>
<li>Microindel detection in short-read sequence data. Krawitz et al, Bioinformatics 26(6):722-9, 2010. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/26/6/722.long">Full Text</a></li>
<li>vipR: variant identification in pooled DNA using R. Altmann et al., Bioinformatics 27: i77-i84, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3117388">PubMedCentral</a></li>
<li>Geoseq: a tool for dissecting deep-sequencing datasets. Gurtowski et al, BMC Bioinformatics 11:506, 2010. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2972303/">PubMedCentral</a> [<em>Geoseq is a web service that allows searching deep sequencing datasets with a reference sequence of a gene of interest.</em>]</li>
<li>Detecting and annotating genetic variations using the HugeSeq pipeline. Lam et al, Nature Biotechnology 30:226 - 229, 2012 <a class="reference external" href="http://www.nature.com/nbt/journal/v30/n3/full/nbt.2134.html">Publisher Website</a>, <a class="reference external" href="http://hugeseq.snyderlab.org/">Home Page</a></li>
<li>Genome-wide LORE1 retrotransposon mutagenesis and high-throughput insertion detection in Lotus japonicus. Urbański et al, Plant J 64:731-741, 2012. <a class="reference external" href="http://onlinelibrary.wiley.com/doi/10.1111/j.1365-313X.2011.04827.x/abstract">Publisher Website</a> [<em>This paper describes a 2-dimensional pooling strategy with barcoding to allow use of Illumina sequencing to screen for retrotransposon insertion mutations, and includes a software package called FSTpoolit for analysis of the resulting sequence reads.</em>]</li>
<li>Reproducibility of variant calls in replicate next-generation sequencing experiments. Qi et al., PLoS One 10: e0119230, 2015 <a class="reference external" href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0119230">Full Text</a></li>
</ul>
</div></blockquote>
<ul class="simple">
<li>Genotyping by sequencing
+ Genome-wide genetic marker discovery and genotyping using next-generation sequencing. Davey et al., Nat Rev Genet 12(7):499-510, 2011 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21681211">PubMed</a> [<em>A review of methods available at the time.</em>]
+ A robust, simple genotyping-by-sequencing (GBS) approach for high diversity species. Elshire et al., PLoS One 6(5):e19379, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3087801">Full Text</a>
+ Development of high-density genetic maps for barley and wheat using a novel two-enzyme genotyping-by-sequencing approach. Poland et al., PLoS One 7(2): e32253, 2012. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3289635/">Full Text</a>
+ Double digest RADseq: an inexpensive method for de novo SNP discovery and genotyping in model and non-model species. Peterson et al, PLoS One 7(5):e37135, . 2012. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3365034/">Full Text</a>
+ Imputation of unordered markers and the impact on genomic selection accuracy. Rutkowski et al, G3 3(3):427-39, 2013. <a class="reference external" href="http://www.g3journal.org/content/3/3/427.long">Full Text</a>
+ Diversity Arrays Technology (DArT) and next-generation sequencing combined: genome-wide, high-throughput, highly informative genotyping for molecular breeding of Eucalyptus. Sansaloni et al., BMC Proceedings 5(Suppl 7):P54, 2011 <a class="reference external" href="http://www.biomedcentral.com/1753-6561/5/S7/P54">Full Text</a>
+ High-throughput genotyping by whole-genome resequencing. Huang et al., Genome Res 19(6):1068-76, 2009. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2694477">Full Text</a>
+ Multiplexed shotgun genotyping for rapid and efficient genetic mapping. Andolfatto et al. Genome Res 21(4):610-7, 2011. <a class="reference external" href="http://genome.cshlp.org/content/21/4/610.long">Full Text</a></li>
<li>Restriction-site Associated DNA (RAD) markers
+ Rapid SNP discovery and genetic mapping using sequenced RAD markers. Baird et al, PLoS One 3(10):e3376, 2008 <a class="reference external" href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0003376">Full Text</a>
+ Linkage mapping and comparative genomics using next-generation RAD sequencing of a non-model organism. Baxter et al., PLoS One 6(4):e19315, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3082572">Full Text</a>
+ Genome evolution and meiotic maps by massively parallel DNA sequencing: spotted gar, an outgroup for the teleost genome duplication. Amores et al, Genetics 188(4):799-808, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21828280">PubMed</a>
+ Construction and application for QTL analysis of a Restriction-site Associated DNA (RAD) linkage map in barley. Chutimanitsakun et al, BMC Genomics 4; 12:4, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3023751">Full Text</a>
+ RAD tag sequencing as a source of SNP markers in Cynara cardunculus L. Scaglione et al., BMC Genomics 13:3, 2012. <cite>Full Text &lt;http://www.biomedcentral.com/1471-2164/13/3&gt;</cite>
+ Paired-end RAD-seq for de novo assembly and marker design without available reference. Willing et al., Bioinformatics 27(16):2187-93, 2011. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/27/16/2187.long">Publisher Website</a>
+ Local de novo assembly of RAD paired-end contigs using short sequencing reads. Etter et al., PLOS ONE 6(4): e18561, 2011. <a class="reference external" href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0018561">Full Text</a>
+ Stacks: building and genotyping loci de novo from short-read sequences. Catchen et al., G3: Genes, Genomes, Genetics, 1:171-182, 2011. <a class="reference external" href="http://creskolab.uoregon.edu/stacks/">Home Page</a>
+ Rainbow: an integrated tool for efficient clustering and assembling RAD-seq reads. Chong et al, Bioinformatics 28(21):2732-7, 2012. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/28/21/2732.long">Publisher Website</a>
+ UK RAD Sequencing Wiki page, with bibliography and RADTools software download <a class="reference external" href="https://www.wiki.ed.ac.uk/display/RADSequencing/Home">Home Page</a></li>
</ul>
</div>
<div class="section" id="population-genomics">
<h2>Population Genomics<a class="headerlink" href="#population-genomics" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>PGDspider: an automated data conversion tool for connecting population genetics and genomics programs. Lischer &amp; Excoffier, Bioinformatics 28: 298-299, 2012 <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/28/2/298.full">Publisher Website</a></li>
</ul>
</div>
<div class="section" id="workspace-environments">
<h2>Workspace environments<a class="headerlink" href="#workspace-environments" title="Permalink to this headline">¶</a></h2>
<ul>
<li><p class="first">Papers
+ Galaxy: a comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences. Goecks et al, Genome Biol 11(8):R86, 2010 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2945788">PubMedCentral</a>
+ Galaxy Cloudman: Delivering compute clusters. BMC Bioinformatics 11(Suppl. 12):S4, 2010 <a class="reference external" href="http://www.biomedcentral.com/content/pdf/1471-2105-11-S12-S4.pdf">Full Text</a>
+ <a class="reference external" href="http://www.broadinstitute.org/gsa/wiki/index.php/The_Genome_Analysis_Toolkit">The Genome Analysis Toolkit</a>: a MapReduce framework for analyzing next-generation DNA sequencing data. McKenna et al, Genome Res 20(9):1297-303, 2010. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928508">PubMedCentral</a>
+ A framework for variation discovery and genotyping using next-generation DNA sequencing data. DePristo et al., Nat Genet 43(5):491-8, 2011. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21478889">PubMed</a></p>
</li>
<li><p class="first">Online resources
+ The <a class="reference external" href="http://cran.r-project.org/">R statistical computing</a> environment includes <a class="reference external" href="http://www.bioconductor.org/">Bioconductor</a>, a specialized set of tools for analysis of microarray and high-throughput sequencing data. Introductory materials from on-line or short workshops are widely available online; examples are <a class="reference external" href="http://bioconductor.org/help/course-materials/2012/Evomics2012/Bioconductor-tutorial.pdf">Evomics2012 Bioconductor Tutorial</a>, and <a class="reference external" href="http://bcb.dfci.harvard.edu/%7Eaedin/courses/Bioconductor/">Intro to Bioconductor</a>. Materials from an advanced course on high-throughput genetic data analysis are at <a class="reference external" href="http://bioconductor.org/help/course-materials/2012/SeattleFeb2012/">Seattle 2012 materials</a>. Thomas Girke of UC-Riverside has written a very complete set of manuals describing the use of R and Bioconductor for analysis of genomic datasets, available at <a class="reference external" href="http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual">R and Bioconductor Manuals</a>.</p>
<p><a class="reference external" href="http://cran.r-project.org/manuals.html">Manuals</a> and contributed <a class="reference external" href="http://cran.r-project.org/other-docs.html">documentation</a> for R are available at the R-project.org website, and video tutorials are also available on Youtube; those posted by Tutorlol are brief, clear, and to the point.</p>
<dl class="docutils">
<dt>Materials from a series of mini-courses in R taught in 2010 at UCLA are available:</dt>
<dd><ul class="first last simple">
<li><a class="reference external" href="http://scc.stat.ucla.edu/page_attachments/0000/0141/10S-basicR.pdf">Intro to programming and graphics</a></li>
<li><a class="reference external" href="http://scc.stat.ucla.edu/page_attachments/0000/0143/S10_RProgII.pdf">Data manipulation and functions</a></li>
<li><a class="reference external" href="http://scc.stat.ucla.edu/page_attachments/0000/0185/Graphics_course.pdf">Graphics for exploratory data analysis</a></li>
<li><a class="reference external" href="http://scc.stat.ucla.edu/page_attachments/0000/0147/20100503_IntroStats.pdf">Introductory statistics</a></li>
<li><a class="reference external" href="http://scc.stat.ucla.edu/page_attachments/0000/0188/reg_R_1_09S_slides.pdf">Linear regression</a></li>
</ul>
</dd>
</dl>
<p><a class="reference external" href="http://a-little-book-of-r-for-bioinformatics.readthedocs.org/en/latest/">A Little Book of R for Bioinformatics</a> is an on-line resource with information and exercises to provide practice in bioinformatics analysis of DNA sequences and other biological data in R. Many books on specific topics in R programming are also available through Amazon or other vendors.</p>
</li>
<li><p class="first">Cloud computing resources
+ The case for cloud computing in genome informatics. Lincoln Stein, Genome Biol. 11(5):207, 2010 <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20441614">Pubmed</a>
+ Galaxy Cloudman: delivering cloud compute clusters. Afgan et al, BMC Bioinformatics 11(Suppl 12):S4, 2010 <a class="reference external" href="http://www.biomedcentral.com/1471-2105/11/S12/S4">Full Text</a>
+ <a class="reference external" href="http://cloudbiolinux.com/">CloudBioLinux</a> is an open-source project that provides a bioinformatics Linux system for cloud computing, pre-configured with a variety of software tools installed and ready to use.
+ A <a class="reference external" href="https://github.com/chapmanb/cloudbiolinux/blob/master/doc/intro/gettingStarted_CloudBioLinux.pdf?raw=true">tutorial</a> on getting started with CloudBioLinux on the Amazon Web Services Elastic Compute Cloud (EC2)
+ <a class="reference external" href="http://userwww.service.emory.edu/%7Eeafgan/content/ppt/EnisAfgan_BOSC_2010.pdf">Deploying Galaxy on the Cloud</a> slides from a presentation by Enis Afgan (Emory University) at the Bioinformatics Open Source Conference in Boston, July 2010
+ A <a class="reference external" href="http://screencast.g2.bx.psu.edu/cloud/">screencast</a> that provides a step-by-step guide to starting a Galaxy cluster in the EC2 environment
+ A <cite>webpage &lt;https://bitbucket.org/galaxy/galaxy-central/wiki/cloud&gt;</cite> that has the same information in text form, and is the basis for the screencast
+ The iPlant Collaborative, an NSF-funded project to create computational resources for plant biology research, provides access to cloud computing resources through <a class="reference external" href="http://www.iplantcollaborative.org/discover/atmosphere">Atmosphere</a>
+ SeqWare Query Engine: storing and searching sequence data in the cloud. OConnor et al, BMC Bioinformatics 11(Suppl 12):S2, 2010 <a class="reference external" href="http://www.biomedcentral.com/1471-2105/11/S12/S2">Full Text</a>
+ An overview of the Hadoop/MapReduce/HBase framework and its current applications in bioinformatics. Taylor, BMC Bioinformatics 11(Suppl 12):S1, 2010 <a class="reference external" href="http://www.biomedcentral.com/1471-2105/11/S12/S1">Full Text</a></p>
</li>
<li><p class="first">Links to Linux command-line tutorials and resources
Tutorials for AWK, a powerful tool for handling data tables
+ A set of <a class="reference external" href="http://people.bu.edu/scottm/AWK.NOTES">awk notes</a> from Boston University
+ Bruce Barnett’s <a class="reference external" href="http://www.grymoire.com/Unix/Awk.html">awk tutorial</a>
+ Greg Goebel’s <a class="reference external" href="http://www.vectorsite.net/tsawk.html">awk tutorial</a>
+ <a class="reference external" href="http://teaching.software-carpentry.org/2013/01/16/1433/">Executing an awk command from R</a> to simplify data exploratory analysis, from Lex Nederbragt
Tutorials for bash shell scripting
+ A <a class="reference external" href="http://www.linuxconfig.org/bash-scripting-tutorial">tutorial</a> at linuxconfig.org
+ A <a class="reference external" href="http://www.hypexr.org/bash_tutorial.php">Getting Started With Bash</a> tutorial at hypexr.org
+ Mendel Cooper’s <a class="reference external" href="http://tldp.org/LDP/abs/html/">Advanced Bash Shell-Scripting Guide</a>
Tutorials for sed, the command-line stream editor
+ A <a class="reference external" href="http://www.panix.com/%7Eelflord/unix/sed.html">tutorial</a> at Rutgers
+ Peter Krumins claims to have the <a class="reference external" href="http://www.catonmat.net/blog/worlds-best-introduction-to-sed/">World’s Best Introduction to Sed</a>; take a look and judge for yourself.
+ Bruce Barnett’s <a class="reference external" href="http://www.grymoire.com/Unix/Sed.html">sed tutorial</a>.</p>
</li>
</ul>
</div>
<div class="section" id="links-to-other-useful-sites">
<h2>Links to other useful sites<a class="headerlink" href="#links-to-other-useful-sites" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>The <a class="reference external" href="http://seqanswers.com/">SEQanswers</a> online community has forums on several topics related to sequencing; the bioinformatics forum is the most active.</li>
<li>The SEQanswers <a class="reference external" href="http://seqanswers.com/wiki/Software">Software Wiki</a> is a list of software for analysis of sequencing data</li>
<li><a class="reference external" href="http://biostar.stackexchange.com/">Biostar</a> is another online community for questions and answers on bioinformatics and computational genomics.</li>
<li>Information on file formats used by the University of California - Santa Cruz Genome Browser is on the <a class="reference external" href="http://genome.ucsc.edu/FAQ/FAQformat">FAQ list</a></li>
<li>A manual for the Integrated Genome Browser visualization tool is <a class="reference external" href="http://wiki.transvar.org/confluence/display/igbman/Home">here</a></li>
<li>Course materials for a short course entitled <a class="reference external" href="http://bioconductor.org/help/course-materials/2010/SeattleIntro/">Introduction to R and Bioconductor</a>, held in Seattle in Dec 2010</li>
<li><a class="reference external" href="http://great.stanford.edu/">Genomic Regions Enrichment of Annotations Tool</a> - A web service to test for over-representation of specific ontology categories among genes near ChIP-seq peaks</li>
<li>Ben Langmead, author of several tools for sequence analysis, has made <a class="reference external" href="https://github.com/BenLangmead/comp-genomics-class">course materials</a> for a class in Computational Genomics available on Github.</li>
<li>An open-source book called <a class="reference external" href="http://readiab.org/book/latest/">Introduction to Applied Bioinformatics</a> has chapters on sequence alignment approaches and algorithms, for those interested in more detail about how that works.</li>
<li><a class="reference external" href="http://www.animalgenome.org/bioinfo/resources/nextgensoft.html">Next-gen-seq software</a> - a list of software packages, both commercial and open-source, related to analysis of deep sequencing datasets</li>
<li><a class="reference external" href="http://www.cbcb.umd.edu/software/">Software</a> from the Center for Bioinformatics and Computational Biology, University of Maryland - many useful programs, all open-source</li>
<li><a class="reference external" href="http://bioinformatics.psb.ugent.be/plaza/">PLAZA</a>: a comparative genomics resource to study gene and genome evolution in plants; described by Proost et al, Plant Cell 21:3718, 2010 <a class="reference external" href="http://www.plantcell.org/content/21/12/3718.full">Full Text</a></li>
<li>The European Bioinformatics Institute provides tools <a class="reference external" href="http://www.ebi.ac.uk/Tools/rcloud/">ArrayExpressHTS and R-Cloud</a> for analysis of transcriptome data</li>
</ul>
<p>Last modified 18 December 2018.
Edits by <a class="reference external" href="https://github.com/rwhetten">Ross Whetten</a>, <a class="reference external" href="https://github.com/wkohlway">Will Kohlway</a>, &amp; <a class="reference external" href="https://github.com/amalgamaria">Maria Adonay</a>.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Useful Publications and Websites</a><ul>
<li><a class="reference internal" href="#global-overview-books-and-papers">Global overview books and papers</a></li>
<li><a class="reference internal" href="#library-construction-and-experimental-design">Library construction and experimental design</a></li>
<li><a class="reference internal" href="#data-formats-data-management-and-alignment-software-tools">Data formats, data management, and alignment software tools</a></li>
<li><a class="reference internal" href="#data-quality-assessment-filtering-and-correction">Data quality assessment, filtering, and correction</a></li>
<li><a class="reference internal" href="#de-novo-assembly">De novo assembly</a></li>
<li><a class="reference internal" href="#chromatin-analysis">Chromatin analysis</a></li>
<li><a class="reference internal" href="#transcriptome-analysis">Transcriptome analysis</a></li>
<li><a class="reference internal" href="#comparing-genomes-and-assemblies-variant-detection">Comparing genomes and assemblies; variant detection</a></li>
<li><a class="reference internal" href="#population-genomics">Population Genomics</a></li>
<li><a class="reference internal" href="#workspace-environments">Workspace environments</a></li>
<li><a class="reference internal" href="#links-to-other-useful-sites">Links to other useful sites</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="topics.html"
                        title="previous chapter">Topics</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="HPC.html"
                        title="next chapter">Using the HPC for Bioinformatics Computing</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/readings.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="HPC.html" title="Using the HPC for Bioinformatics Computing"
             >next</a> |</li>
        <li class="right" >
          <a href="topics.html" title="Topics"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BIT 815 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Ross Whetten.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.2.
    </div>
  </body>
</html>