
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Transcriptome Assembly &#8212; BIT 815 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/pyramid.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Neuton&amp;subset=latin" type="text/css" media="screen" charset="utf-8" />
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nobile:regular,italic,bold,bolditalic&amp;subset=latin" type="text/css" media="screen" charset="utf-8" />
<!--[if lte IE 6]>
<link rel="stylesheet" href="_static/ie6.css" type="text/css" media="screen" charset="utf-8" />
<![endif]-->

  </head><body>

    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="index.html">BIT 815 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <a class="reference external image-reference" href="http://www.ncsu.edu"><img alt="_images/NC_State.gif" src="_images/NC_State.gif" /></a>
<a class="reference external image-reference" href="http://www.cnr.ncsu.edu/fer/"><img alt="_images/FERlogo.png" src="_images/FERlogo.png" /></a>
<div class="section" id="transcriptome-assembly">
<h1>Transcriptome Assembly<a class="headerlink" href="#transcriptome-assembly" title="Permalink to this headline">¶</a></h1>
<div class="section" id="global-overview">
<h2>Global Overview<a class="headerlink" href="#global-overview" title="Permalink to this headline">¶</a></h2>
<p>Multiple options are available for reconstruction of the sequences of RNA transcripts based on analysis of complementary DNA copies, depending on whether a high-quality reference genome assembly is available. If so, reference-guided transcriptome assembly using a splice-aware sequencer aligner followed by resolution of alternative splicing variants and export of consensus transcript sequences can be powerful (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3334321/">Trapnell et al., 2012</a>). If there is no reference genome assembly, or the available assembly is fragmented and poorly-annotated, a de-novo assembly of putative transcripts from short-read sequences is one alternative. The availability of long-read sequencing methods such as Oxford Nanopore Technologies and Pacific Biosciences has opened a third alternative, which is to obtain full-length sequences of cDNAs and report those sequences without the requirement for assembly at all (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4075632/">Sharon et al., 2013</a>; <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0777-z">Bolisetty et al., 2015</a>; <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0729-7">Minoche et al., 2015</a>).</p>
</div>
<div class="section" id="objective">
<h2>Objective<a class="headerlink" href="#objective" title="Permalink to this headline">¶</a></h2>
<p>The objective of this session is to introduce participants to methods for assembly of a transcriptome, or a collection of RNA transcripts present in a given RNA sample.  The RNA-seq data used in the differential gene analysis exercise will be used for this exercise as well, as those datasets are small enough to run on systems with only 8 Gb RAM.</p>
</div>
<div class="section" id="description">
<h2>Description<a class="headerlink" href="#description" title="Permalink to this headline">¶</a></h2>
<p>De-novo assembly of short DNA sequence reads into contiguous full-length copies of RNA transcripts is a complex process, because there are multiple sources of variation in read coverage and in read sequences. Read coverage is proportional to transcript abundance, but can also be affected by biases due to library construction  methods and biological factors. A key factor leading to variation in transcript sequences, and therefore variation in read sequences, is alternative splicing of transcripts from the same transcription unit into different mature forms. Another factor affecting read sequences is the increased error rate of RNA polymerases relative to DNA polymerases - RNA sequences are inherently noisier than DNA sequences. Trinity (Grabherr et al, 2011) is a widely-used suite of programs for de-novo assembly of RNA-seq data; the <a class="reference external" href="https://cs.wellesley.edu/%7Ebtjaden/Rockhopper/">Rockhopper</a> program for assembly of bacterial RNA-seq data is described in a recent report (Tjaden, 2015). One key difference between these two programs is the amount of memory required - the recommended amount of RAM for Trinity is 1 Gb RAM per million reads in the dataset, while Rockhopper can carry out assembly of datasets with hundreds of millions of reads using 2 Gb of RAM.</p>
<p>Construction of a de Bruijn graph is one approach used to solve the computational problem of assembling short DNA sequence reads into accurate contigs that reflect the transcripts present in the original RNA sample used for library construction. A key difference between Trinity and Rockhopper in the amount of memory required is that Rockhopper builds an index of sequence contigs as they are assembled, so that the entire de Bruijn graph of all k-mers present in the full set of reads does not have to be held in memory for the entire process of assembly. <a class="reference external" href="http://www4.ncsu.edu/~rosswhet/BIT815/Overview/Resources/RNA_seq_data_class.pdf">Slides</a> from <a class="reference external" href="https://www.linkedin.com/in/lilian-matallana-21704474/">Lilian Matallana</a>’s lecture on de Bruijn graphs and their use in sequence assembly programs describe these different strategies in more detail.</p>
</div>
<div class="section" id="key-facts">
<h2>Key Facts<a class="headerlink" href="#key-facts" title="Permalink to this headline">¶</a></h2>
<p>Paired-end sequencing reads are useful for assembly of eukaryotic transcriptomes, because the information they provide about the positions of sequences relative to each other within a transcript is valuable for correct assembly of alternatively-spliced transcripts. In bacteria, splicing is less frequent, alternative splicing can be less of a concern, and single-end reads can often be assembled into a reasonably accurate transcriptome. If alternative splicing is not of interest, then single-end reads are useful for RNA-seq analysis of eukaryotic transcriptomes as well - now that read-lengths of  100 to 150 nt are readily available, paired-end reads add little additional information regarding levels of gene expression.</p>
</div>
<div class="section" id="exercise-reference-guided-assembly">
<h2>Exercise - reference-guided assembly<a class="headerlink" href="#exercise-reference-guided-assembly" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>The Arabidopsis thaliana RNA-seq dataset used for the analysis of differential gene expression includes six sequence files in the /data/AtRNAseq folder on the VCL Biostar_DNASeq Linux system.</li>
<li>The HiSat2 aligner is already installed on the VCL system. If you want to install it on another computer, an executable binary can be downloaded from the link under the Releases heading on the right side of the program <a class="reference external" href="http://ccb.jhu.edu/software/hisat2/index.shtml">home page</a>. This program is the successor to Bowtie and Tophat, the original programs developed for reference-guided sequence assembly.</li>
<li>Create a folder for this exercise, and unpack the Atchromo5.fasta.gz reference sequence from the /data/AtRNAseq folder into that directory. Use the hisat2-build program (from the hisat2-2.0.5 directory) to build an alignment index from the uncompressed fasta-format sequence data.</li>
<li>Unpack the three control-sample RNA-seq read files (named c1, c2 and c3) from the /data/AtRNAseq directory to a single fastq file, using the gzip -cd command with file globs.</li>
<li>Align the fastq-format sequence reads to the reference sequence using the hisat2 program (using the –dta option), and pipe the output to samtools1.3 view to convert the SAM output into BAM, then to samtools1.3 sort to sort the output BAM data and save it to a file.</li>
<li>Download the Linux x86_64 binary version of the StringTie program from the link under the Obtaining and installing StringTie heading on the program <a class="reference external" href="http://ccb.jhu.edu/software/stringtie/index.shtml">home page</a>. Unpack the zip archive in your /home/lubuntu directory.</li>
<li>Execute the stringtie program using the sorted BAM file as input.</li>
</ul>
</div>
<div class="section" id="exercise-de-novo-assembly">
<h2>Exercise - de-novo assembly<a class="headerlink" href="#exercise-de-novo-assembly" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>The <em>Arabidopsis thaliana</em> RNA-seq dataset can also be used for de-novo assembly, although RAM is a limiting factor on instances of the VCL Biostar_DNASeq machine image.</li>
<li><a class="reference external" href="https://cs.wellesley.edu/%7Ebtjaden/Rockhopper/">Rockhopper</a> can be downloaded to the home directory of a VCL instance and run from the command line  - for some reason the GUI version would not save the file of assembled transcripts when I tested it. All six files of RNA-seq data are from the same accession of Arabidopsis, so they can all be concatenated into a single file and provided as input to Rockhopper.</li>
<li>A file of <em>Arabidopsis thaliana</em> RNA sequences (inferred from gene models in the TAIR 10 genome assembly) is available in the /data/AtRNAseq folder. The assembled transcripts can be compared with these predicted transcripts as a means of evaluating how good a job the Rockhopper assembler (which is designed for assembly of bacterial RNA-seq datasets) does with the plant RNA-seq data.</li>
</ul>
</div>
<div class="section" id="additional-resources">
<h2>Additional Resources<a class="headerlink" href="#additional-resources" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>One strategy for reducing the amount of RAM required for transcriptome assembly by the Trinity software package is to carry out “digital normalization” of the RNA-seq dataset - this means adjusting the numbers of reads in the dataset to ensure more uniform representation of both abundant and rare transcripts, while removing sequencing errors. A detailed <a class="reference external" href="http://khmer-protocols.readthedocs.io/en/v0.8.4/mrnaseq/index.html">exercise</a> is available, which uses AWS cloud computing instances to provide sufficient computing power to process a real dataset.</li>
<li>Analysis of Next Generation Sequencing data (ANGUS) is a workshop series on high-throughput sequence data analysis; the <a class="reference external" href="https://angus.readthedocs.io/en/2017/toc.html">2017 workshop</a> includes an <a class="reference external" href="https://angus.readthedocs.io/en/2017/assembly-trinity.html">exercise</a> on transcriptome assembly with Trinity using cloud computing resources.</li>
<li>Rana et al., 2016. Comparison of de-novo transcriptome assemblers and k-mer strategies using the killifish, Fundulus heteroclitus. PLoS One 11: e0153104. <a class="reference external" href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0153104">Full text</a></li>
<li>Boley et al., 2014. Genome-guided transcript assembly by integrative analysis of RNA sequence data. Nature Biotechnology 32: 341-346. <a class="reference external" href="http://www.nature.com/nbt/journal/v32/n4/full/nbt.2850.html">Publisher Website</a></li>
<li>Grabherr et al, 2011. Full-length transcriptome assembly from RNA-Seq data without a reference genome. Nature Biotechnology 29:644 - 652. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3571712/">PubMed Central</a></li>
<li>Tjaden, B. (2015) De novo genome assembly of bacterial transcriptomes from RNA-seq data. Genome Biology 16:1 <a class="reference external" href="http://genomebiology.com/2015/16/1/1">Full text</a></li>
<li>Gnerre S, et al. (2011) High-quality draft assemblies of mammalian genomes from massively parallel sequence data. Proc Natl Acad Sci USA 108:1513–1518. <cite>PubMedCentral &lt;http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3029755/&gt;</cite></li>
<li>Salzberg S, et al. (2012) GAGE: A critical evaluation of genome assemblies and assembly algorithms. Genome Research 22:557–567. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3290791">PubMedCentral</a> This paper describes a set of experiments comparing different assembly programs on four genomes, and provides useful insights into the challenges of genome assembly.</li>
<li>Magoc T and Salzberg S. (2011) FLASH: Fast Length Adjustment of Short Reads to improve genome assemblies. Bioinformatics 27:2957–2963. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3198573/">PubMedCentral</a> This paper describes a software tool for joining paired-end reads obtained from DNA fragments short enough that the reads overlap at the ends. This is reported to improve the quality of assemblies created from the joined reads. An outline of an exercise with the FLASH assembler is available: <a class="reference external" href="p:%5Cfer%5CChristmas%20Tree%20Genetics%20Program%5COther%20Files%5CBIT815_WebpageCode%5CBIT815%5CDocuments%20for%20classes%5CWeek%204">FLASH_exercise.docx</a></li>
<li>Pevzner PA, et al. (2001) An Eulerian path approach to DNA fragment assembly. PNAS 98:9748-9753. <a class="reference external" href="http://www.pnas.org/content/98/17/9748.full">Full Text</a></li>
</ul>
<p>Last modified 18 December 2018.
Edits by <a class="reference external" href="https://github.com/rwhetten">Ross Whetten</a>, <a class="reference external" href="https://github.com/wkohlway">Will Kohlway</a>, &amp; <a class="reference external" href="https://github.com/amalgamaria">Maria Adonay</a>.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Transcriptome Assembly</a><ul>
<li><a class="reference internal" href="#global-overview">Global Overview</a></li>
<li><a class="reference internal" href="#objective">Objective</a></li>
<li><a class="reference internal" href="#description">Description</a></li>
<li><a class="reference internal" href="#key-facts">Key Facts</a></li>
<li><a class="reference internal" href="#exercise-reference-guided-assembly">Exercise - reference-guided assembly</a></li>
<li><a class="reference internal" href="#exercise-de-novo-assembly">Exercise - de-novo assembly</a></li>
<li><a class="reference internal" href="#additional-resources">Additional Resources</a></li>
</ul>
</li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/transcriptome-assembly.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="index.html">BIT 815 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Ross Whetten.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.2.
    </div>
  </body>
</html>